{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import subprocess\n",
    "import os\n",
    "import glob\n",
    "import sys\n",
    "import pwd\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy\n",
    "import scipy.stats as stats\n",
    "from sklearn.decomposition import PCA\n",
    "import pickle\n",
    "import re\n",
    "import math\n",
    "import plotnine as pn\n",
    "import seaborn as sns\n",
    "import glob\n",
    "import subprocess\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "import seaborn as sns\n",
    "from numpy import log\n",
    "from collections import defaultdict,Counter\n",
    "import re\n",
    "import statsmodels\n",
    "import warnings\n",
    "import statsmodels.stats.multitest as smm\n",
    "import json\n",
    "\n",
    "from pathlib import Path\n",
    "#from midas import m\n",
    "#from datetime import datetime, date\n",
    "#import altair as alt\n",
    "\n",
    "# gh specific\n",
    "sys.path.append('/ghsfa/projects/pharma/shared_scripts/')\n",
    "import bibp.functions as functions\n",
    "import hgvslib.pHGVS as pHGVS  ## do a pull on the hgvslib git repo\n",
    "\n",
    "# !pip install statannotations==0.4.0\n",
    "# #!pip uninstall statannotations --yes\n",
    "# from statannotations.Annotator import Annotator\n",
    "\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "pd.options.display.max_rows = None\n",
    "pd.options.display.max_columns=None\n",
    "pd.options.display.max_colwidth=None\n",
    "work_dir=\"/ghsfa/projects/pharma/projects/sirius_pharma/hazhang_projects/CRC_methylation_caller_result_check_01202025\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data preocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch is 42\n",
      "\n",
      "Total unique sample size of the Infinity batch is 42\n",
      "\n",
      "Total unique patient sample size of the Infinity batch is 32\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226443</td>\n",
       "      <td>A122644301</td>\n",
       "      <td>140-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.003628</td>\n",
       "      <td>58.33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1203278</td>\n",
       "      <td>B00831229</td>\n",
       "      <td>121-009</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.146274</td>\n",
       "      <td>20.15</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/7/24</td>\n",
       "      <td>11/6/24</td>\n",
       "      <td>12/2/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1204902</td>\n",
       "      <td>B00832399</td>\n",
       "      <td>141-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.009578</td>\n",
       "      <td>195.64</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/8/24</td>\n",
       "      <td>11/7/24</td>\n",
       "      <td>11/25/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1204331</td>\n",
       "      <td>B00831858</td>\n",
       "      <td>141-006</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.304769</td>\n",
       "      <td>21.90</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/8/24</td>\n",
       "      <td>11/7/24</td>\n",
       "      <td>11/25/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1238988</td>\n",
       "      <td>A123898801</td>\n",
       "      <td>104-023</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.430250</td>\n",
       "      <td>26.06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/12/24</td>\n",
       "      <td>12/11/24</td>\n",
       "      <td>1/9/25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "0  XTX101-0102-001               NaN    A1226443  A122644301    140-001   \n",
       "1  XTX101-0102-001               NaN    A1203278   B00831229    121-009   \n",
       "2  XTX101-0102-001               NaN    A1204902   B00832399    141-005   \n",
       "3  XTX101-0102-001               NaN    A1204331   B00831858    141-006   \n",
       "4  XTX101-0102-001               NaN    A1238988  A123898801    104-023   \n",
       "\n",
       "                  Cancertype  \\\n",
       "0  Colorectal adenocarcinoma   \n",
       "1  Colorectal adenocarcinoma   \n",
       "2  Colorectal adenocarcinoma   \n",
       "3  Colorectal adenocarcinoma   \n",
       "4  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "1  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "2  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "3  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "4  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "0       SUCCESS             NaN       Detected                 Detected   \n",
       "1       SUCCESS             NaN       Detected                 Detected   \n",
       "2       SUCCESS             NaN       Detected                 Detected   \n",
       "3       SUCCESS             NaN       Detected                 Detected   \n",
       "4       SUCCESS             NaN       Detected                 Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "0                 0.003628           58.33              NaN   \n",
       "1                 0.146274           20.15              4.5   \n",
       "2                 0.009578          195.64              4.5   \n",
       "3                 0.304769           21.90              4.0   \n",
       "4                 0.430250           26.06              NaN   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \n",
       "0                  NaN      11/30/24       11/27/24      12/13/24  \n",
       "1                  NaN       11/7/24        11/6/24       12/2/24  \n",
       "2                  NaN       11/8/24        11/7/24      11/25/24  \n",
       "3                  NaN       11/8/24        11/7/24      11/25/24  \n",
       "4                  NaN      12/12/24       12/11/24        1/9/25  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Readin the Infinity batch\n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20241114_AstraZeneca_SOW202_Batch1_80Samples_InfinityReport.csv\", header = 1) \n",
    "\n",
    "# Jill's batches\n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/customers/Pfizer/Infinity/PFE_SOW51core_Infinity/PFE_SOW51core_Infinity_03/data/reports/20250121_Pfizer_SOW51core_Batch3_18Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250117_Pfizer_SOW46Amendment1TP1_Batch9_1Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250117_Pfizer_SOW46Amendment1TP1_Batch10_3Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250120_Pfizer_SOW46Amendment1TP3_Batch9_20Samples_G360Report.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250120_Pfizer_SOW46Amendment1TP3_Batch10_8Samples_G360Report.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250117_Pfizer_SOW46Amendment1TP5_Batch6_11Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250120_Pfizer_SOW46Amendment1TP5_Batch7_33Samples_InfinityReport.csv\", header = 1) \n",
    "\n",
    "\n",
    "# Mingyang's batches\n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/data/infinity_report_warehouse/20241220_GuardantHealthIST_IST01_Batch1_405Samples_InfinityReport.csv\", header = 1) \n",
    "\n",
    "# Haiyang's batch\n",
    "df_ori = pd.read_csv(\"/ghsfa/projects/pharma/projects/sirius_pharma/hazhang_projects/XTX_TFv2_update_Batch1to40_01162025/df_xtx_Sirius161.csv\", header = 0) \n",
    "\n",
    "\n",
    "print(\"Total sample size of the Infinity batch is \" + str(df_ori.shape[0]) + \"\\n\") \n",
    "print(\"Total unique sample size of the Infinity batch is \" + str(df_ori.GHSampleID.nunique()) + \"\\n\")\n",
    "print(\"Total unique patient sample size of the Infinity batch is \" + str(df_ori.Patient_ID.nunique()) + \"\\n\")\n",
    "\n",
    "df_ori.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226443</td>\n",
       "      <td>A122644301</td>\n",
       "      <td>140-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.003628</td>\n",
       "      <td>58.33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1203278</td>\n",
       "      <td>B00831229</td>\n",
       "      <td>121-009</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.146274</td>\n",
       "      <td>20.15</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/7/24</td>\n",
       "      <td>11/6/24</td>\n",
       "      <td>12/2/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1204902</td>\n",
       "      <td>B00832399</td>\n",
       "      <td>141-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.009578</td>\n",
       "      <td>195.64</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/8/24</td>\n",
       "      <td>11/7/24</td>\n",
       "      <td>11/25/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1204331</td>\n",
       "      <td>B00831858</td>\n",
       "      <td>141-006</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.304769</td>\n",
       "      <td>21.90</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/8/24</td>\n",
       "      <td>11/7/24</td>\n",
       "      <td>11/25/24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1238988</td>\n",
       "      <td>A123898801</td>\n",
       "      <td>104-023</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.430250</td>\n",
       "      <td>26.06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/12/24</td>\n",
       "      <td>12/11/24</td>\n",
       "      <td>1/9/25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "0  XTX101-0102-001               NaN    A1226443  A122644301    140-001   \n",
       "1  XTX101-0102-001               NaN    A1203278   B00831229    121-009   \n",
       "2  XTX101-0102-001               NaN    A1204902   B00832399    141-005   \n",
       "3  XTX101-0102-001               NaN    A1204331   B00831858    141-006   \n",
       "4  XTX101-0102-001               NaN    A1238988  A123898801    104-023   \n",
       "\n",
       "                  Cancertype  \\\n",
       "0  Colorectal adenocarcinoma   \n",
       "1  Colorectal adenocarcinoma   \n",
       "2  Colorectal adenocarcinoma   \n",
       "3  Colorectal adenocarcinoma   \n",
       "4  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "1  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "2  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "3  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "4  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "0       SUCCESS             NaN       Detected                 Detected   \n",
       "1       SUCCESS             NaN       Detected                 Detected   \n",
       "2       SUCCESS             NaN       Detected                 Detected   \n",
       "3       SUCCESS             NaN       Detected                 Detected   \n",
       "4       SUCCESS             NaN       Detected                 Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "0                 0.003628           58.33              NaN   \n",
       "1                 0.146274           20.15              4.5   \n",
       "2                 0.009578          195.64              4.5   \n",
       "3                 0.304769           21.90              4.0   \n",
       "4                 0.430250           26.06              NaN   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \n",
       "0                  NaN      11/30/24       11/27/24      12/13/24  \n",
       "1                  NaN       11/7/24        11/6/24       12/2/24  \n",
       "2                  NaN       11/8/24        11/7/24      11/25/24  \n",
       "3                  NaN       11/8/24        11/7/24      11/25/24  \n",
       "4                  NaN      12/12/24       12/11/24        1/9/25  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove duplication\n",
    "df_dedup = df_ori.drop_duplicates(subset = \"GHSampleID\")\n",
    "df_dedup.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch after merging runid is 42\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1210253</td>\n",
       "      <td>A121025301</td>\n",
       "      <td>104-017</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.346847</td>\n",
       "      <td>42.92</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/14/24</td>\n",
       "      <td>11/13/24</td>\n",
       "      <td>11/25/24</td>\n",
       "      <td>241121_A01810_0520_BHT73GDSXC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226429</td>\n",
       "      <td>A122642901</td>\n",
       "      <td>146-004</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.119741</td>\n",
       "      <td>37.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226435</td>\n",
       "      <td>A122643501</td>\n",
       "      <td>146-003</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.057710</td>\n",
       "      <td>77.38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/29/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226443</td>\n",
       "      <td>A122644301</td>\n",
       "      <td>140-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.003628</td>\n",
       "      <td>58.33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226445</td>\n",
       "      <td>A122644501</td>\n",
       "      <td>140-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.012115</td>\n",
       "      <td>49.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Study_ID Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "21  XTX101-0102-001               NaN    A1210253  A121025301    104-017   \n",
       "13  XTX101-0102-001               NaN    A1226429  A122642901    146-004   \n",
       "18  XTX101-0102-001               NaN    A1226435  A122643501    146-003   \n",
       "0   XTX101-0102-001               NaN    A1226443  A122644301    140-001   \n",
       "28  XTX101-0102-001               NaN    A1226445  A122644501    140-005   \n",
       "\n",
       "                   Cancertype  \\\n",
       "21  Colorectal adenocarcinoma   \n",
       "13  Colorectal adenocarcinoma   \n",
       "18  Colorectal adenocarcinoma   \n",
       "0   Colorectal adenocarcinoma   \n",
       "28  Colorectal adenocarcinoma   \n",
       "\n",
       "                                              Tumor_stage  Visit_name  \\\n",
       "21  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "13                                              Stage III         NaN   \n",
       "18  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "0   Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "28  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "   Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "21       SUCCESS             NaN       Detected                 Detected   \n",
       "13       SUCCESS             NaN       Detected                 Detected   \n",
       "18       SUCCESS             NaN       Detected                 Detected   \n",
       "0        SUCCESS             NaN       Detected                 Detected   \n",
       "28       SUCCESS             NaN       Detected                 Detected   \n",
       "\n",
       "    Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "21                 0.346847           42.92              4.5   \n",
       "13                 0.119741           37.01              NaN   \n",
       "18                 0.057710           77.38              NaN   \n",
       "0                  0.003628           58.33              NaN   \n",
       "28                 0.012115           49.20              NaN   \n",
       "\n",
       "    Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "21                  NaN      11/14/24       11/13/24      11/25/24   \n",
       "13                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "18                  NaN      11/30/24       11/29/24      12/13/24   \n",
       "0                   NaN      11/30/24       11/27/24      12/13/24   \n",
       "28                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "\n",
       "                            runid  \n",
       "21  241121_A01810_0520_BHT73GDSXC  \n",
       "13  241205_A00277_1180_AHTGM7DSXC  \n",
       "18  241205_A00277_1180_AHTGM7DSXC  \n",
       "0   241205_A00277_1180_AHTGM7DSXC  \n",
       "28  241205_A00277_1180_AHTGM7DSXC  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pull \"runid\" from ghdb\n",
    "df_1 = df_dedup.copy()\n",
    "\n",
    "# Define the IDs you want to query for\n",
    "ids = tuple(df_1['GHSampleID'].unique())\n",
    "\n",
    "# Generate the SQL query string\n",
    "query = \"SELECT * FROM gh_sample WHERE run_sample_id IN {}\".format(ids)\n",
    "\n",
    "# Get the dataframe with \"runid\"\n",
    "sample_df = functions.run_sql_query(query)\n",
    "sample_df = sample_df.rename(columns={\"run_sample_id\": \"GHSampleID\"})\n",
    "\n",
    "# Found some sample has duplicated runid which might due to re-run. So only the latest FC runs were saved.\n",
    "# Extract the date from runid and convert to datetime format for accurate comparison\n",
    "sample_df['run_date'] = pd.to_datetime(sample_df['runid'].str.extract(r'(\\d{6})')[0], format='%y%m%d')\n",
    "df_1 = df_1.merge(sample_df[['GHSampleID', 'runid', 'run_date']], on='GHSampleID', how=\"left\")\n",
    "df_1 = df_1.sort_values(by=['GHSampleID', 'run_date'], ascending=[True, False]).drop_duplicates(subset='GHSampleID', keep='first')\n",
    "df_1 = df_1.drop(columns='run_date')\n",
    "\n",
    "print(\"Total sample size of the Infinity batch after merging runid is \" + str(df_1.shape[0])) \n",
    "df_1.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch after merging runid is 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Alt-NoAlt</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>Indel_type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Mut_aa</th>\n",
       "      <th>Mut_nt</th>\n",
       "      <th>Mut_cdna</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>Percentage</th>\n",
       "      <th>Splice_effect</th>\n",
       "      <th>Somatic_status</th>\n",
       "      <th>Molecular_consequence</th>\n",
       "      <th>Fusion_chrom_b</th>\n",
       "      <th>Fusion_gene_b</th>\n",
       "      <th>Fusion_position_a</th>\n",
       "      <th>Fusion_position_b</th>\n",
       "      <th>Direction_a</th>\n",
       "      <th>Direction_b</th>\n",
       "      <th>Downstream_gene</th>\n",
       "      <th>Copy_number</th>\n",
       "      <th>CNV_type</th>\n",
       "      <th>COSMIC</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVarID</th>\n",
       "      <th>Functional_impact</th>\n",
       "      <th>Mutant_allele_status</th>\n",
       "      <th>Mol_count</th>\n",
       "      <th>Max_percentage</th>\n",
       "      <th>Alleletype</th>\n",
       "      <th>HRD_score</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>TMB_score</th>\n",
       "      <th>TMB_category</th>\n",
       "      <th>MSI_High</th>\n",
       "      <th>cfDNA_ng</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Practice_name</th>\n",
       "      <th>Physician_name</th>\n",
       "      <th>runid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C4221015</td>\n",
       "      <td>13431002_Cycle 1 Day 1</td>\n",
       "      <td>A1218829</td>\n",
       "      <td>B00888193</td>\n",
       "      <td>13431002</td>\n",
       "      <td>Cycle 1 Day 1</td>\n",
       "      <td>1</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PIN1</td>\n",
       "      <td>19</td>\n",
       "      <td>9958804.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>A124S</td>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>c.370G&gt;T</td>\n",
       "      <td>NM_006221.4</td>\n",
       "      <td>51.12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>germline</td>\n",
       "      <td>missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs138056875</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5935.0</td>\n",
       "      <td>5.85</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.060326</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.056315</td>\n",
       "      <td>22.78</td>\n",
       "      <td>High</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>22.85</td>\n",
       "      <td>3.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11/16/2024</td>\n",
       "      <td>06/20/2024</td>\n",
       "      <td>01/17/2025</td>\n",
       "      <td>CRC</td>\n",
       "      <td>Not Applicable</td>\n",
       "      <td>Not Applicable</td>\n",
       "      <td>250106_A00277_1208_BHWTHFDSXC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Study_ID       Customer_SampleID GHRequestID GHSampleID  Patient_ID  \\\n",
       "0  C4221015  13431002_Cycle 1 Day 1    A1218829  B00888193    13431002   \n",
       "\n",
       "      Visit_name  Alt-NoAlt Sample_status  Sample_comment Variant_type  \\\n",
       "0  Cycle 1 Day 1          1       SUCCESS             NaN          SNV   \n",
       "\n",
       "  Indel_type  Gene Chromosome   Position  Exon Mut_aa Mut_nt  Mut_cdna  \\\n",
       "0        NaN  PIN1         19  9958804.0   3.0  A124S    G>T  c.370G>T   \n",
       "\n",
       "    Transcript  Percentage Splice_effect Somatic_status Molecular_consequence  \\\n",
       "0  NM_006221.4       51.12           NaN       germline              missense   \n",
       "\n",
       "   Fusion_chrom_b  Fusion_gene_b  Fusion_position_a  Fusion_position_b  \\\n",
       "0             NaN            NaN                NaN                NaN   \n",
       "\n",
       "   Direction_a  Direction_b  Downstream_gene  Copy_number  CNV_type COSMIC  \\\n",
       "0          NaN          NaN              NaN          NaN       NaN    NaN   \n",
       "\n",
       "         dbSNP ClinVar  ClinVarID Functional_impact  Mutant_allele_status  \\\n",
       "0  rs138056875     NaN        NaN               NaN                   NaN   \n",
       "\n",
       "   Mol_count  Max_percentage  Alleletype  HRD_score Tumor_methylation_status  \\\n",
       "0     5935.0            5.85         NaN   0.060326                 Detected   \n",
       "\n",
       "   Tumor_methylation_score  TMB_score TMB_category      MSI_High  cfDNA_ng  \\\n",
       "0                 0.056315      22.78         High  Not Detected     22.85   \n",
       "\n",
       "   Plasma_ml_input  Plasma_ml_remaining Received_date Bloodcoll_date  \\\n",
       "0              3.5                  0.0    11/16/2024     06/20/2024   \n",
       "\n",
       "  Reported_date Cancertype   Practice_name  Physician_name  \\\n",
       "0    01/17/2025        CRC  Not Applicable  Not Applicable   \n",
       "\n",
       "                           runid  \n",
       "0  250106_A00277_1208_BHWTHFDSXC  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Optional step!!!\n",
    "# # If you only have one sample in the batch, run this cell alternatively instead of from above one\n",
    "\n",
    "# # Pull \"runid\" from ghdb\n",
    "# df_1 = df_dedup.copy()\n",
    "\n",
    "# # Define the IDs you want to query for\n",
    "# ids = tuple(df_1['GHSampleID'].unique())\n",
    "\n",
    "# # If there's only one ID, format it without a comma\n",
    "# if len(ids) == 1:\n",
    "#     ids = f\"('{ids[0]}')\"\n",
    "# else:\n",
    "#     ids = str(ids)\n",
    "\n",
    "# # Generate the SQL query string\n",
    "# query = f\"SELECT * FROM gh_sample WHERE run_sample_id IN {ids}\"\n",
    "\n",
    "# # Get the dataframe with \"runid\"\n",
    "# sample_df = functions.run_sql_query(query)\n",
    "# sample_df = sample_df.rename(columns={\"run_sample_id\": \"GHSampleID\"})\n",
    "\n",
    "# # Found some sample has duplicated runid which might due to re-run. So only the latest FC runs were saved.\n",
    "# # Extract the date from runid and convert to datetime format for accurate comparison\n",
    "# sample_df['run_date'] = pd.to_datetime(sample_df['runid'].str.extract(r'(\\d{6})')[0], format='%y%m%d')\n",
    "# df_1 = df_1.merge(sample_df[['GHSampleID', 'runid', 'run_date']], on='GHSampleID', how=\"left\")\n",
    "# df_1 = df_1.sort_values(by=['GHSampleID', 'run_date'], ascending=[True, False]).drop_duplicates(subset='GHSampleID', keep='first')\n",
    "# df_1 = df_1.drop(columns='run_date')\n",
    "\n",
    "# print(\"Total sample size of the Infinity batch after merging runid is \" + str(df_1.shape[0]))\n",
    "# df_1.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch with fc_dir is 42\n",
      "\n",
      "Total sample size of the Infinity batch with NO fc_dir is 0\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "      <th>fc_dir</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1210253</td>\n",
       "      <td>A121025301</td>\n",
       "      <td>104-017</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.346847</td>\n",
       "      <td>42.92</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/14/24</td>\n",
       "      <td>11/13/24</td>\n",
       "      <td>11/25/24</td>\n",
       "      <td>241121_A01810_0520_BHT73GDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241121_A01810_0520_BHT73GDSXC/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226429</td>\n",
       "      <td>A122642901</td>\n",
       "      <td>146-004</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.119741</td>\n",
       "      <td>37.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226435</td>\n",
       "      <td>A122643501</td>\n",
       "      <td>146-003</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.057710</td>\n",
       "      <td>77.38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/29/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226443</td>\n",
       "      <td>A122644301</td>\n",
       "      <td>140-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.003628</td>\n",
       "      <td>58.33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226445</td>\n",
       "      <td>A122644501</td>\n",
       "      <td>140-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.012115</td>\n",
       "      <td>49.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Study_ID Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "21  XTX101-0102-001               NaN    A1210253  A121025301    104-017   \n",
       "13  XTX101-0102-001               NaN    A1226429  A122642901    146-004   \n",
       "18  XTX101-0102-001               NaN    A1226435  A122643501    146-003   \n",
       "0   XTX101-0102-001               NaN    A1226443  A122644301    140-001   \n",
       "28  XTX101-0102-001               NaN    A1226445  A122644501    140-005   \n",
       "\n",
       "                   Cancertype  \\\n",
       "21  Colorectal adenocarcinoma   \n",
       "13  Colorectal adenocarcinoma   \n",
       "18  Colorectal adenocarcinoma   \n",
       "0   Colorectal adenocarcinoma   \n",
       "28  Colorectal adenocarcinoma   \n",
       "\n",
       "                                              Tumor_stage  Visit_name  \\\n",
       "21  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "13                                              Stage III         NaN   \n",
       "18  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "0   Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "28  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "   Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "21       SUCCESS             NaN       Detected                 Detected   \n",
       "13       SUCCESS             NaN       Detected                 Detected   \n",
       "18       SUCCESS             NaN       Detected                 Detected   \n",
       "0        SUCCESS             NaN       Detected                 Detected   \n",
       "28       SUCCESS             NaN       Detected                 Detected   \n",
       "\n",
       "    Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "21                 0.346847           42.92              4.5   \n",
       "13                 0.119741           37.01              NaN   \n",
       "18                 0.057710           77.38              NaN   \n",
       "0                  0.003628           58.33              NaN   \n",
       "28                 0.012115           49.20              NaN   \n",
       "\n",
       "    Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "21                  NaN      11/14/24       11/13/24      11/25/24   \n",
       "13                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "18                  NaN      11/30/24       11/29/24      12/13/24   \n",
       "0                   NaN      11/30/24       11/27/24      12/13/24   \n",
       "28                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "\n",
       "                            runid  \\\n",
       "21  241121_A01810_0520_BHT73GDSXC   \n",
       "13  241205_A00277_1180_AHTGM7DSXC   \n",
       "18  241205_A00277_1180_AHTGM7DSXC   \n",
       "0   241205_A00277_1180_AHTGM7DSXC   \n",
       "28  241205_A00277_1180_AHTGM7DSXC   \n",
       "\n",
       "                                                         fc_dir  \n",
       "21  /ghsfa/ivd/flowcentral/241121_A01810_0520_BHT73GDSXC/sirius  \n",
       "13  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius  \n",
       "18  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius  \n",
       "0   /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius  \n",
       "28  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def add_fc_dir_column(df):\n",
    "    \"\"\"\n",
    "    Adds a 'fc_dir' column to the given dataframe, specifying the FC path based on the 'runid' column.\n",
    "\n",
    "    Args:\n",
    "        df (pd.DataFrame): A DataFrame that must contain a 'runid' column. This column is used to\n",
    "                           match directory paths from predefined locations.\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The original dataframe with an added 'fc_dir' column indicating the directory path\n",
    "                      corresponding to each 'runid'. If no directory is found, the path will be an empty string.\n",
    "        pd.DataFrame: A DataFrame containing the rows for which no directory path was found.\n",
    "    \"\"\"\n",
    "    # Define the path\n",
    "    fc_dir_all = \"/ghsfa/ivd/flowcentral\"\n",
    "    fc_dir_omni = \"/ghsfa/projects/omni_v2/flowcells\"\n",
    "    \n",
    "    # Initialize a list to store rows where no BIP output was found\n",
    "    rows_no_found = []\n",
    "    \n",
    "    # Initialize the new column in the DataFrame\n",
    "    df['fc_dir'] = ''\n",
    "    \n",
    "    # Iterate over the DataFrame rows\n",
    "    for index, row in df.iterrows():\n",
    "        runid = row[\"runid\"]\n",
    "        \n",
    "        # Find original folder\n",
    "        pattern_ori = f\"{fc_dir_all}/*{runid}*\"\n",
    "        folders = glob.glob(pattern_ori)\n",
    "        if folders:\n",
    "            for folder in folders:\n",
    "                primary_fc_dir = folder\n",
    "                # Check if there is a 'sirius' folder under the primary fc_dir. This is for Sirius 1.5 folder structure change.\n",
    "                if os.path.exists(os.path.join(primary_fc_dir, \"sirius\")):\n",
    "                    primary_fc_dir = os.path.join(primary_fc_dir, \"sirius\")\n",
    "                df.loc[index, 'fc_dir'] = primary_fc_dir\n",
    "        else:\n",
    "            # Search in the alternative directory if no folder found in the first directory\n",
    "            pattern_omni = f\"{fc_dir_omni}/*{runid}*\"\n",
    "            folders_omni = glob.glob(pattern_omni)\n",
    "            if folders_omni:\n",
    "                df.loc[index, 'fc_dir'] = folders_omni[0]\n",
    "            else:\n",
    "                # Append the row to the list if no folder is found in either location\n",
    "                rows_no_found.append(row)\n",
    "    \n",
    "    # Create a DataFrame with rows where folders were not found in any sources\n",
    "    df_no_fc_dir = pd.DataFrame(rows_no_found)\n",
    "    \n",
    "    return df, df_no_fc_dir\n",
    "\n",
    "# Generate the new dataframe \n",
    "df_2, df_no_found = add_fc_dir_column(df_1)\n",
    "\n",
    "print(\"Total sample size of the Infinity batch with fc_dir is \" + str(df_2.shape[0]) + \"\\n\") \n",
    "print(\"Total sample size of the Infinity batch with NO fc_dir is \" + str(df_no_found.shape[0]) + \"\\n\") \n",
    "\n",
    "df_2.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate S3 MB CRC caller outputs (Sirius 1.5.0)\n",
    "###### Git clone Sirius-1.5.0_Release_Branch for region_file and control_file files for different S3 MB callers\n",
    "###### Branch https://github.com/guardant/bip-products/blob/Sirius-1.5.0-RLS/Sirius_LDT_Soft/Sirius_LDT_Soft.yaml#L610-L623\n",
    "###### Cloned Sirius-1.5.0_Release_Branch dir: /home/hazhang/projects/Repo/bip-products/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.0.3\n"
     ]
    }
   ],
   "source": [
    "# Make sure update the epicaller to 2.0.3 or above version\n",
    "import epicaller\n",
    "print(epicaller.__version__)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "import epicaller\n",
    "import epicaller.methylation.methyl_caller\n",
    "\n",
    "def mb_crc_caller_sirius150_wrapper_slurm(sample_meta_df, workdir):\n",
    "    region_file='/home/hazhang/projects/Repo/bip-products/models/Sirius/mafband_crc.classification_space.tsv'\n",
    "    control_region_file='/home/hazhang/projects/Repo/bip-products/models/Sirius/mafband_crc.positive_control_regions.tsv'\n",
    "    mr_model_file='/home/hazhang/projects/Repo/bip-products/models/Sirius/mafband_crc.threshold_set_to_zero.tsv'\n",
    "\n",
    "    jobs_dir = f\"{workdir}/MB_CRC_log\"\n",
    "    output_dir = f\"{workdir}/MB_CRC_output\"\n",
    "\n",
    "    # Check and create directories if not exists\n",
    "    if not os.path.exists(jobs_dir):\n",
    "        os.makedirs(jobs_dir)\n",
    "\n",
    "    if not os.path.exists(output_dir):\n",
    "        os.makedirs(output_dir)\n",
    "\n",
    "    for _, row in sample_meta_df.iterrows():\n",
    "        run_sample_id = row[\"GHSampleID\"]\n",
    "        fc_dir = row[\"fc_dir\"]\n",
    "        flowcell_id = row[\"runid\"]\n",
    "        mbd_hyper_molecules = f\"{fc_dir}/{run_sample_id}/{run_sample_id}.mbd_hyper_molecules.tsv\"\n",
    "        \n",
    "        # Check if the molecule file exists\n",
    "        if not os.path.exists(mbd_hyper_molecules):\n",
    "            continue\n",
    "\n",
    "        job_file = f\"{jobs_dir}/MB_call_{run_sample_id}.job\"\n",
    "        shebang_content = (f\"#!/bin/bash\\n\"\n",
    "                           f\"#SBATCH --job-name=MB_call_{run_sample_id}\\n\"\n",
    "                           f\"#SBATCH --nodes=1\\n\"\n",
    "                           f\"#SBATCH --ntasks=1\\n\"\n",
    "                           f\"#SBATCH --cpus-per-task=1\\n\"\n",
    "                           f\"#SBATCH --mem=30gb\\n\"\n",
    "                           f\"#SBATCH --partition=dev\\n\"\n",
    "                           f\"#SBATCH --time=00:10:00\\n\"\n",
    "                           f\"#SBATCH --output={jobs_dir}/MB_call_{run_sample_id}.log\\n\\n\")\n",
    "\n",
    "        with open(job_file, \"w\") as job_output:\n",
    "            job_output.write(shebang_content)\n",
    "            job_output.write(\"python3 -m epicaller.methylation.methyl_caller \" + \n",
    "                             f\"--hyper-molecule-file {mbd_hyper_molecules} \" + \n",
    "                             f\"--region-file {region_file} \" +\n",
    "                             f\"--control-region-file {control_region_file} \" + \n",
    "                             f\"--model-file {mr_model_file} \" +\n",
    "                             \"--model-name crc --min-frag-len 120 --max-frag-len 240 --min-cg-count 12 --max-cg-count 30 --cutsite-count-cutoff 1 --min-mapq 60 \" + \n",
    "                             f\"--sample-id {run_sample_id} --flowcell-id {flowcell_id} --outdir {output_dir}/\")\n",
    "            # print(job_output)    \n",
    "        os.system(f\"sbatch {job_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 97846581\n",
      "Submitted batch job 97846582\n",
      "Submitted batch job 97846583\n",
      "Submitted batch job 97846584\n",
      "Submitted batch job 97846586\n",
      "Submitted batch job 97846587\n",
      "Submitted batch job 97846589\n",
      "Submitted batch job 97846591\n",
      "Submitted batch job 97846593\n",
      "Submitted batch job 97846594\n",
      "Submitted batch job 97846596\n",
      "Submitted batch job 97846597\n",
      "Submitted batch job 97846599\n",
      "Submitted batch job 97846601\n",
      "Submitted batch job 97846602\n",
      "Submitted batch job 97846604\n",
      "Submitted batch job 97846606\n",
      "Submitted batch job 97846608\n",
      "Submitted batch job 97846610\n",
      "Submitted batch job 97846612\n",
      "Submitted batch job 97846614\n",
      "Submitted batch job 97846615\n",
      "Submitted batch job 97846617\n",
      "Submitted batch job 97846619\n",
      "Submitted batch job 97846621\n",
      "Submitted batch job 97846623\n",
      "Submitted batch job 97846624\n",
      "Submitted batch job 97846626\n",
      "Submitted batch job 97846627\n",
      "Submitted batch job 97846629\n",
      "Submitted batch job 97846631\n",
      "Submitted batch job 97846633\n",
      "Submitted batch job 97846635\n",
      "Submitted batch job 97846636\n",
      "Submitted batch job 97846638\n",
      "Submitted batch job 97846639\n",
      "Submitted batch job 97846641\n",
      "Submitted batch job 97846646\n",
      "Submitted batch job 97846649\n",
      "Submitted batch job 97846650\n",
      "Submitted batch job 97846652\n",
      "Submitted batch job 97846653\n"
     ]
    }
   ],
   "source": [
    "# Generate the MB S3 CRC caller outputs (Sirius 1.5.0)\n",
    "mb_crc_caller_sirius150_wrapper_slurm(df_2, f\"{work_dir}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "      <th>fc_dir</th>\n",
       "      <th>MB_CRC_call_sirius150_version</th>\n",
       "      <th>MB_CRC_epi_score_sirius150_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1210253</td>\n",
       "      <td>A121025301</td>\n",
       "      <td>104-017</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.346847</td>\n",
       "      <td>42.92</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/14/24</td>\n",
       "      <td>11/13/24</td>\n",
       "      <td>11/25/24</td>\n",
       "      <td>241121_A01810_0520_BHT73GDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241121_A01810_0520_BHT73GDSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>49.155813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226429</td>\n",
       "      <td>A122642901</td>\n",
       "      <td>146-004</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.119741</td>\n",
       "      <td>37.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>37.811887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226435</td>\n",
       "      <td>A122643501</td>\n",
       "      <td>146-003</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.057710</td>\n",
       "      <td>77.38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/29/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>41.845025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226443</td>\n",
       "      <td>A122644301</td>\n",
       "      <td>140-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.003628</td>\n",
       "      <td>58.33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>17.901997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226445</td>\n",
       "      <td>A122644501</td>\n",
       "      <td>140-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.012115</td>\n",
       "      <td>49.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>26.892025</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Study_ID Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "21  XTX101-0102-001               NaN    A1210253  A121025301    104-017   \n",
       "13  XTX101-0102-001               NaN    A1226429  A122642901    146-004   \n",
       "18  XTX101-0102-001               NaN    A1226435  A122643501    146-003   \n",
       "0   XTX101-0102-001               NaN    A1226443  A122644301    140-001   \n",
       "28  XTX101-0102-001               NaN    A1226445  A122644501    140-005   \n",
       "\n",
       "                   Cancertype  \\\n",
       "21  Colorectal adenocarcinoma   \n",
       "13  Colorectal adenocarcinoma   \n",
       "18  Colorectal adenocarcinoma   \n",
       "0   Colorectal adenocarcinoma   \n",
       "28  Colorectal adenocarcinoma   \n",
       "\n",
       "                                              Tumor_stage  Visit_name  \\\n",
       "21  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "13                                              Stage III         NaN   \n",
       "18  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "0   Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "28  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "   Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "21       SUCCESS             NaN       Detected                 Detected   \n",
       "13       SUCCESS             NaN       Detected                 Detected   \n",
       "18       SUCCESS             NaN       Detected                 Detected   \n",
       "0        SUCCESS             NaN       Detected                 Detected   \n",
       "28       SUCCESS             NaN       Detected                 Detected   \n",
       "\n",
       "    Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "21                 0.346847           42.92              4.5   \n",
       "13                 0.119741           37.01              NaN   \n",
       "18                 0.057710           77.38              NaN   \n",
       "0                  0.003628           58.33              NaN   \n",
       "28                 0.012115           49.20              NaN   \n",
       "\n",
       "    Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "21                  NaN      11/14/24       11/13/24      11/25/24   \n",
       "13                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "18                  NaN      11/30/24       11/29/24      12/13/24   \n",
       "0                   NaN      11/30/24       11/27/24      12/13/24   \n",
       "28                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "\n",
       "                            runid  \\\n",
       "21  241121_A01810_0520_BHT73GDSXC   \n",
       "13  241205_A00277_1180_AHTGM7DSXC   \n",
       "18  241205_A00277_1180_AHTGM7DSXC   \n",
       "0   241205_A00277_1180_AHTGM7DSXC   \n",
       "28  241205_A00277_1180_AHTGM7DSXC   \n",
       "\n",
       "                                                         fc_dir  \\\n",
       "21  /ghsfa/ivd/flowcentral/241121_A01810_0520_BHT73GDSXC/sirius   \n",
       "13  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "18  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "0   /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "28  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "\n",
       "   MB_CRC_call_sirius150_version  MB_CRC_epi_score_sirius150_version  \n",
       "21                      Detected                           49.155813  \n",
       "13                      Detected                           37.811887  \n",
       "18                      Detected                           41.845025  \n",
       "0                       Detected                           17.901997  \n",
       "28                      Detected                           26.892025  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Retrieve MB CRC score and CRC call of Sirius 1.5.0 result\n",
    "# df_3 = df_2.copy()\n",
    "\n",
    "# # Prep some columns for comparison\n",
    "# df_3[\"MB_CRC_call_sirius150_version\"] = np.nan\n",
    "# df_3[\"MB_CRC_epi_score_sirius150_version\"] = np.nan\n",
    "\n",
    "# # # Optional for pulling current BIP run version (only works BIP prior to Sirius 1.6.x; adding /sirius subfolder for Sirius 1.6.x)\n",
    "# # df_3[\"MB_CRC_call_BIP_run_version\"] = np.nan\n",
    "# # df_3[\"MB_CRC_BIP_score_BIP_run_version\"] = np.nan\n",
    "\n",
    "# mb_caller_result_folder = f\"{work_dir}/MB_CRC_output\"\n",
    "\n",
    "\n",
    "# for index, crc_row in df_3.iterrows():\n",
    "#     run_sample_id = crc_row[\"GHSampleID\"]\n",
    "#     msre_mr_file = f\"{mb_caller_result_folder}/{run_sample_id}.msre_crc_call.hdr.tsv\" \n",
    "#     msre_df = pd.read_table(msre_mr_file)\n",
    "\n",
    "#     # Retrieve the Sirius 1.5.0 results\n",
    "#     df_3.at[index, \"MB_CRC_epi_score_sirius150_version\"] = msre_df.loc[0, \"methyl_score\"]\n",
    "#     df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\" if msre_df.loc[0, \"methyl_call\"] == 1 else \"Not Detected\"\n",
    "\n",
    "#     # # Optional step\n",
    "#     # # Retrieve the current BIP run results\n",
    "#     # fc_dir = crc_row[\"fc_dir\"]\n",
    "#     # msre_mr_BIP_file = f\"{fc_dir}/{run_sample_id}/{run_sample_id}.msre_all_call.hdr.tsv\" \n",
    "#     # msre_BIP_df = pd.read_table(msre_mr_BIP_file)\n",
    "\n",
    "#     # msre_BIP_df = msre_BIP_df[msre_BIP_df['model_name'] == 'CRC'].reset_index(drop=True)\n",
    "#     # if not msre_BIP_df.empty:\n",
    "#     #     df_3.at[index, \"MB_CRC_BIP_score\"] = msre_BIP_df.loc[0, \"methyl_score\"]\n",
    "#     #     df_3.at[index, \"MB_CRC_BIP_call\"] = \"Detected\" if msre_BIP_df.loc[0, \"methyl_call\"] == 1 else \"Not Detected\"\n",
    "\n",
    "# df_3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "      <th>fc_dir</th>\n",
       "      <th>MB_CRC_call_sirius150_version</th>\n",
       "      <th>MB_CRC_epi_score_sirius150_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1210253</td>\n",
       "      <td>A121025301</td>\n",
       "      <td>104-017</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.346847</td>\n",
       "      <td>42.92</td>\n",
       "      <td>4.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/14/24</td>\n",
       "      <td>11/13/24</td>\n",
       "      <td>11/25/24</td>\n",
       "      <td>241121_A01810_0520_BHT73GDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241121_A01810_0520_BHT73GDSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>28.955813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226429</td>\n",
       "      <td>A122642901</td>\n",
       "      <td>146-004</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.119741</td>\n",
       "      <td>37.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>17.611887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226435</td>\n",
       "      <td>A122643501</td>\n",
       "      <td>146-003</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.057710</td>\n",
       "      <td>77.38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/29/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>21.645025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226443</td>\n",
       "      <td>A122644301</td>\n",
       "      <td>140-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.003628</td>\n",
       "      <td>58.33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-2.298003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1226445</td>\n",
       "      <td>A122644501</td>\n",
       "      <td>140-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.012115</td>\n",
       "      <td>49.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/30/24</td>\n",
       "      <td>11/27/24</td>\n",
       "      <td>12/13/24</td>\n",
       "      <td>241205_A00277_1180_AHTGM7DSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>6.692025</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Study_ID Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "21  XTX101-0102-001               NaN    A1210253  A121025301    104-017   \n",
       "13  XTX101-0102-001               NaN    A1226429  A122642901    146-004   \n",
       "18  XTX101-0102-001               NaN    A1226435  A122643501    146-003   \n",
       "0   XTX101-0102-001               NaN    A1226443  A122644301    140-001   \n",
       "28  XTX101-0102-001               NaN    A1226445  A122644501    140-005   \n",
       "\n",
       "                   Cancertype  \\\n",
       "21  Colorectal adenocarcinoma   \n",
       "13  Colorectal adenocarcinoma   \n",
       "18  Colorectal adenocarcinoma   \n",
       "0   Colorectal adenocarcinoma   \n",
       "28  Colorectal adenocarcinoma   \n",
       "\n",
       "                                              Tumor_stage  Visit_name  \\\n",
       "21  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "13                                              Stage III         NaN   \n",
       "18  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "0   Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "28  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "   Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "21       SUCCESS             NaN       Detected                 Detected   \n",
       "13       SUCCESS             NaN       Detected                 Detected   \n",
       "18       SUCCESS             NaN       Detected                 Detected   \n",
       "0        SUCCESS             NaN       Detected                 Detected   \n",
       "28       SUCCESS             NaN       Detected                 Detected   \n",
       "\n",
       "    Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "21                 0.346847           42.92              4.5   \n",
       "13                 0.119741           37.01              NaN   \n",
       "18                 0.057710           77.38              NaN   \n",
       "0                  0.003628           58.33              NaN   \n",
       "28                 0.012115           49.20              NaN   \n",
       "\n",
       "    Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "21                  NaN      11/14/24       11/13/24      11/25/24   \n",
       "13                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "18                  NaN      11/30/24       11/29/24      12/13/24   \n",
       "0                   NaN      11/30/24       11/27/24      12/13/24   \n",
       "28                  NaN      11/30/24       11/27/24      12/13/24   \n",
       "\n",
       "                            runid  \\\n",
       "21  241121_A01810_0520_BHT73GDSXC   \n",
       "13  241205_A00277_1180_AHTGM7DSXC   \n",
       "18  241205_A00277_1180_AHTGM7DSXC   \n",
       "0   241205_A00277_1180_AHTGM7DSXC   \n",
       "28  241205_A00277_1180_AHTGM7DSXC   \n",
       "\n",
       "                                                         fc_dir  \\\n",
       "21  /ghsfa/ivd/flowcentral/241121_A01810_0520_BHT73GDSXC/sirius   \n",
       "13  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "18  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "0   /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "28  /ghsfa/ivd/flowcentral/241205_A00277_1180_AHTGM7DSXC/sirius   \n",
       "\n",
       "   MB_CRC_call_sirius150_version  MB_CRC_epi_score_sirius150_version  \n",
       "21                      Detected                           28.955813  \n",
       "13                      Detected                           17.611887  \n",
       "18                      Detected                           21.645025  \n",
       "0                       Detected                           -2.298003  \n",
       "28                      Detected                            6.692025  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Retrieve MB CRC score and CRC call of Sirius 1.5.0 result (IMPORTANT NOTES: only work for late-stage CRC samples and run with Sirius 1.6.1)\n",
    "df_3 = df_2.copy()\n",
    "\n",
    "# Prep some columns for comparison\n",
    "df_3[\"MB_CRC_call_sirius150_version\"] = np.nan\n",
    "df_3[\"MB_CRC_epi_score_sirius150_version\"] = np.nan\n",
    "\n",
    "mb_caller_result_folder = f\"{work_dir}/MB_CRC_output\"\n",
    "\n",
    "for index, crc_row in df_3.iterrows():\n",
    "    run_sample_id = crc_row[\"GHSampleID\"]\n",
    "    msre_mr_file = f\"{mb_caller_result_folder}/{run_sample_id}.msre_crc_call.hdr.tsv\" \n",
    "    msre_df = pd.read_table(msre_mr_file)\n",
    "\n",
    "    # Retrieve and recalculate the methylation score (subtracting 20.2 as epicaller used -20.2 (98% spec) as cutoff)\n",
    "    original_score = msre_df.loc[0, \"methyl_score\"]\n",
    "    crc_methyl_score = original_score - 20.2\n",
    "    df_3.at[index, \"MB_CRC_epi_score_sirius150_version\"] = crc_methyl_score\n",
    "\n",
    "    # Retrieve tumor methylation score\n",
    "    tf_score = crc_row[\"Tumor_methylation_score\"]\n",
    "\n",
    "    # Apply the detection rules of Sirius 1.6.1\n",
    "    if crc_methyl_score >= -21.36903: \n",
    "        df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "    elif (crc_methyl_score >= -22.50785) and (tf_score >= 0.0005):\n",
    "        df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "    else:\n",
    "        df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Not Detected\"\n",
    "\n",
    "\n",
    "df_3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_3 = df_2.copy()\n",
    "\n",
    "# # Prep some columns for comparison\n",
    "# df_3[\"MB_CRC_call_sirius150_version\"] = np.nan\n",
    "# df_3[\"MB_CRC_epi_score_sirius150_version\"] = np.nan\n",
    "\n",
    "# mb_caller_result_folder = f\"{work_dir}/MB_CRC_output\"\n",
    "\n",
    "# for index, crc_row in df_3.iterrows():\n",
    "#     run_sample_id = crc_row[\"GHSampleID\"]\n",
    "#     msre_mr_file = f\"{mb_caller_result_folder}/{run_sample_id}.msre_crc_call.hdr.tsv\" \n",
    "#     msre_df = pd.read_table(msre_mr_file)\n",
    "\n",
    "#     # Retrieve and recalculate the methylation score (subtracting 20.2)\n",
    "#     original_score = msre_df.loc[0, \"methyl_score\"]\n",
    "#     crc_methyl_score = original_score - 20.2\n",
    "#     df_3.at[index, \"MB_CRC_epi_score_sirius150_version\"] = crc_methyl_score\n",
    "\n",
    "#     # Retrieve tumor methylation score\n",
    "#     tf_score = crc_row[\"Tumor_methylation_score\"]\n",
    "\n",
    "#     # Retrieve the BIP version and cancer stage from the dataframe (two columns are fake columns)\n",
    "#     BIP_version = crc_row[\"BIP_version\"]\n",
    "#     Cancer_stage = crc_row[\"Cancer_stage\"]\n",
    "\n",
    "#     # Apply the detection rules based on BIP version and cancer stage\n",
    "#     if BIP_version == \"Sirius 1.6.1\":\n",
    "#         if Cancer_stage == \"late-stage\":\n",
    "#             if crc_methyl_score >= -21.36903:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "#             elif (crc_methyl_score >= -22.50785) and (tf_score >= 0.0005):\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "#             else:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Not Detected\"\n",
    "#         elif Cancer_stage == \"early-stage\":\n",
    "#             if crc_methyl_score >= -20.31064:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "#             else:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Not Detected\"\n",
    "#         else:\n",
    "#             df_3.at[index, \"MB_CRC_call_sirius150_version\"] = np.nan\n",
    "\n",
    "#     elif BIP_version == \"Sirius 1.6.0\":\n",
    "#         if Cancer_stage == \"late-stage\":\n",
    "#             if crc_methyl_score >= -21.249985:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "#             elif (crc_methyl_score >= -22.4522) and (tf_score >= 0.0005):\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "#             else:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Not Detected\"\n",
    "#         elif Cancer_stage == \"early-stage\":\n",
    "#             if crc_methyl_score >= -20.2:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Detected\"\n",
    "#             else:\n",
    "#                 df_3.at[index, \"MB_CRC_call_sirius150_version\"] = \"Not Detected\"\n",
    "#         else:\n",
    "#             df_3.at[index, \"MB_CRC_call_sirius150_version\"] = np.nan\n",
    "#     else:\n",
    "#         df_3.at[index, \"MB_CRC_call_sirius150_version\"] = np.nan\n",
    "\n",
    "# df_3.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total CRC samples with flipped call is 1\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "      <th>fc_dir</th>\n",
       "      <th>MB_CRC_call_sirius150_version</th>\n",
       "      <th>MB_CRC_epi_score_sirius150_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1243649</td>\n",
       "      <td>A124364901</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Detected</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>21.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/18/24</td>\n",
       "      <td>12/17/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241221_A01020_0945_AHWTCMDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241221_A01020_0945_AHWTCMDSXC/sirius</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>-21.906225</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Study_ID Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "20  XTX101-0102-001               NaN    A1243649  A124364901    101-027   \n",
       "\n",
       "         Cancertype                                            Tumor_stage  \\\n",
       "20  Cecum carcinoma  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type   \n",
       "\n",
       "    Visit_name Sample_status  Sample_comment ctDNA_detected  \\\n",
       "20         NaN       SUCCESS             NaN       Detected   \n",
       "\n",
       "   Tumor_methylation_status  Tumor_methylation_score  cfDNA_ng_yield  \\\n",
       "20                 Detected                 0.000143           21.39   \n",
       "\n",
       "    Plasma_ml_input  Plasma_ml_remaining Received_date Bloodcoll_date  \\\n",
       "20              NaN                  NaN      12/18/24       12/17/24   \n",
       "\n",
       "   Reported_date                          runid  \\\n",
       "20        1/2/25  241221_A01020_0945_AHWTCMDSXC   \n",
       "\n",
       "                                                         fc_dir  \\\n",
       "20  /ghsfa/ivd/flowcentral/241221_A01020_0945_AHWTCMDSXC/sirius   \n",
       "\n",
       "   MB_CRC_call_sirius150_version  MB_CRC_epi_score_sirius150_version  \n",
       "20                  Not Detected                          -21.906225  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_flipped_call = df_3[df_3['MB_CRC_call_sirius150_version'] != df_3['Tumor_methylation_status']]\n",
    "print(\"Total CRC samples with flipped call is \" + str(df_flipped_call.shape[0]) + \"\\n\")\n",
    "\n",
    "# Save the data\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20250121_Pfizer_SOW51core_Batch3_18Samples_InfinityReport_flipped_calls.csv\", index = False)\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20241220_GuardantHealthIST_IST01_Batch1_405Samples_InfinityReport_flipped_calls.csv\", index = False)\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20250120_Pfizer_SOW46Amendment1TP3_Batch9_20Samples_G360Report_flipped_calls.csv\", index = False)\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20250120_Pfizer_SOW46Amendment1TP5_Batch7_33Samples_InfinityReport_flipped_calls.csv\", index = False)\n",
    "df_flipped_call.to_csv(f\"{work_dir}/XTX_monitoring_Sirius161_by01242025_flipped_calls.csv\", index = False)\n",
    "\n",
    "df_flipped_call\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "lunartk_v1.1.0",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
