{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import subprocess\n",
    "import os\n",
    "import glob\n",
    "import sys\n",
    "import pwd\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy\n",
    "import scipy.stats as stats\n",
    "from sklearn.decomposition import PCA\n",
    "import pickle\n",
    "import re\n",
    "import math\n",
    "import plotnine as pn\n",
    "import seaborn as sns\n",
    "import glob\n",
    "import subprocess\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "import seaborn as sns\n",
    "from numpy import log\n",
    "from collections import defaultdict,Counter\n",
    "import re\n",
    "import statsmodels\n",
    "import warnings\n",
    "import statsmodels.stats.multitest as smm\n",
    "import json\n",
    "\n",
    "from pathlib import Path\n",
    "#from midas import m\n",
    "#from datetime import datetime, date\n",
    "#import altair as alt\n",
    "\n",
    "# gh specific\n",
    "sys.path.append('/ghsfa/projects/pharma/shared_scripts/')\n",
    "import bibp.functions as functions\n",
    "import hgvslib.pHGVS as pHGVS  ## do a pull on the hgvslib git repo\n",
    "\n",
    "# !pip install statannotations==0.4.0\n",
    "# #!pip uninstall statannotations --yes\n",
    "# from statannotations.Annotator import Annotator\n",
    "\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "pd.options.display.max_rows = None\n",
    "pd.options.display.max_columns=None\n",
    "pd.options.display.max_colwidth=None\n",
    "work_dir=\"/ghsfa/projects/pharma/projects/sirius_pharma/hazhang_projects/CRC_methylation_caller_result_check_01202025\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data preocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch is 7\n",
      "\n",
      "Total unique sample size of the Infinity batch is 7\n",
      "\n",
      "Total unique patient sample size of the Infinity batch is 5\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1277545</td>\n",
       "      <td>A127754501</td>\n",
       "      <td>141-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>19.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/24/25</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>2/11/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1285830</td>\n",
       "      <td>A128583001</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>27.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/31/25</td>\n",
       "      <td>1/30/25</td>\n",
       "      <td>2/19/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1243649</td>\n",
       "      <td>A124364901</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>21.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/18/24</td>\n",
       "      <td>12/17/24</td>\n",
       "      <td>1/2/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1247604</td>\n",
       "      <td>A124760401</td>\n",
       "      <td>146-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>26.64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/21/24</td>\n",
       "      <td>12/19/24</td>\n",
       "      <td>1/2/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1276510</td>\n",
       "      <td>A127651001</td>\n",
       "      <td>140-008</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>33.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>1/22/25</td>\n",
       "      <td>2/11/25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID  Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "0  XTX101-0102-001                NaN    A1277545  A127754501    141-001   \n",
       "1  XTX101-0102-001                NaN    A1285830  A128583001    101-027   \n",
       "2  XTX101-0102-001                NaN    A1243649  A124364901    101-027   \n",
       "3  XTX101-0102-001                NaN    A1247604  A124760401    146-005   \n",
       "4  XTX101-0102-001                NaN    A1276510  A127651001    140-008   \n",
       "\n",
       "                  Cancertype  \\\n",
       "0  Colorectal adenocarcinoma   \n",
       "1            Cecum carcinoma   \n",
       "2            Cecum carcinoma   \n",
       "3  Colorectal adenocarcinoma   \n",
       "4  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "1  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "2  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "3                                              Stage III         NaN   \n",
       "4                                              Stage III         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "0       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "1       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "2       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "3       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "4       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "0                        0           19.86              NaN   \n",
       "1                        0           27.23              NaN   \n",
       "2                        0           21.39              NaN   \n",
       "3                        0           26.64              NaN   \n",
       "4                        0           33.94              NaN   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \n",
       "0                  NaN       1/24/25        1/23/25       2/11/25  \n",
       "1                  NaN       1/31/25        1/30/25       2/19/25  \n",
       "2                  NaN      12/18/24       12/17/24        1/2/25  \n",
       "3                  NaN      12/21/24       12/19/24        1/2/25  \n",
       "4                  NaN       1/23/25        1/22/25       2/11/25  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Readin the Infinity batch\n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20241114_AstraZeneca_SOW202_Batch1_80Samples_InfinityReport.csv\", header = 1) \n",
    "\n",
    "# Jill's batches\n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/customers/Pfizer/Infinity/PFE_SOW51core_Infinity/PFE_SOW51core_Infinity_03/data/reports/20250121_Pfizer_SOW51core_Batch3_18Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250117_Pfizer_SOW46Amendment1TP1_Batch9_1Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250117_Pfizer_SOW46Amendment1TP1_Batch10_3Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250120_Pfizer_SOW46Amendment1TP3_Batch9_20Samples_G360Report.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250120_Pfizer_SOW46Amendment1TP3_Batch10_8Samples_G360Report.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250117_Pfizer_SOW46Amendment1TP5_Batch6_11Samples_InfinityReport.csv\", header = 1) \n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/Infinity/20250120_Pfizer_SOW46Amendment1TP5_Batch7_33Samples_InfinityReport.csv\", header = 1) \n",
    "\n",
    "\n",
    "# Mingyang's batches\n",
    "# df_ori = pd.read_csv(\"/ghsfa/projects/pharma/datawarehouse/data/infinity_report_warehouse/20241220_GuardantHealthIST_IST01_Batch1_405Samples_InfinityReport.csv\", header = 1) \n",
    "\n",
    "# Haiyang's batch\n",
    "df_ori = pd.read_csv(\"/ghsfa/projects/pharma/projects/sirius_pharma/hazhang_projects/XTX_TFv2_update_Batch1to40_01162025/df_xtx_Sirius161_03062025.csv\", header = 0) \n",
    "\n",
    "\n",
    "print(\"Total sample size of the Infinity batch is \" + str(df_ori.shape[0]) + \"\\n\") \n",
    "print(\"Total unique sample size of the Infinity batch is \" + str(df_ori.GHSampleID.nunique()) + \"\\n\")\n",
    "print(\"Total unique patient sample size of the Infinity batch is \" + str(df_ori.Patient_ID.nunique()) + \"\\n\")\n",
    "\n",
    "df_ori.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1277545</td>\n",
       "      <td>A127754501</td>\n",
       "      <td>141-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>19.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/24/25</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>2/11/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1285830</td>\n",
       "      <td>A128583001</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>27.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/31/25</td>\n",
       "      <td>1/30/25</td>\n",
       "      <td>2/19/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1243649</td>\n",
       "      <td>A124364901</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>21.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/18/24</td>\n",
       "      <td>12/17/24</td>\n",
       "      <td>1/2/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1247604</td>\n",
       "      <td>A124760401</td>\n",
       "      <td>146-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>26.64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/21/24</td>\n",
       "      <td>12/19/24</td>\n",
       "      <td>1/2/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1276510</td>\n",
       "      <td>A127651001</td>\n",
       "      <td>140-008</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>33.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>1/22/25</td>\n",
       "      <td>2/11/25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID  Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "0  XTX101-0102-001                NaN    A1277545  A127754501    141-001   \n",
       "1  XTX101-0102-001                NaN    A1285830  A128583001    101-027   \n",
       "2  XTX101-0102-001                NaN    A1243649  A124364901    101-027   \n",
       "3  XTX101-0102-001                NaN    A1247604  A124760401    146-005   \n",
       "4  XTX101-0102-001                NaN    A1276510  A127651001    140-008   \n",
       "\n",
       "                  Cancertype  \\\n",
       "0  Colorectal adenocarcinoma   \n",
       "1            Cecum carcinoma   \n",
       "2            Cecum carcinoma   \n",
       "3  Colorectal adenocarcinoma   \n",
       "4  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "1  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "2  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "3                                              Stage III         NaN   \n",
       "4                                              Stage III         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "0       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "1       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "2       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "3       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "4       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "0                        0           19.86              NaN   \n",
       "1                        0           27.23              NaN   \n",
       "2                        0           21.39              NaN   \n",
       "3                        0           26.64              NaN   \n",
       "4                        0           33.94              NaN   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \n",
       "0                  NaN       1/24/25        1/23/25       2/11/25  \n",
       "1                  NaN       1/31/25        1/30/25       2/19/25  \n",
       "2                  NaN      12/18/24       12/17/24        1/2/25  \n",
       "3                  NaN      12/21/24       12/19/24        1/2/25  \n",
       "4                  NaN       1/23/25        1/22/25       2/11/25  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove duplication\n",
    "df_dedup = df_ori.drop_duplicates(subset = \"GHSampleID\")\n",
    "df_dedup.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch after merging runid is 7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1243649</td>\n",
       "      <td>A124364901</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>21.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/18/24</td>\n",
       "      <td>12/17/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241221_A01020_0945_AHWTCMDSXC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1247604</td>\n",
       "      <td>A124760401</td>\n",
       "      <td>146-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>26.64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/21/24</td>\n",
       "      <td>12/19/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241227_A02019_0258_BHWTKCDSXC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1263902</td>\n",
       "      <td>A126390201</td>\n",
       "      <td>141-006</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>11.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/11/25</td>\n",
       "      <td>1/9/25</td>\n",
       "      <td>2/4/25</td>\n",
       "      <td>250114_A01021_0837_AH27KNDSXF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1276510</td>\n",
       "      <td>A127651001</td>\n",
       "      <td>140-008</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>33.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>1/22/25</td>\n",
       "      <td>2/11/25</td>\n",
       "      <td>250125_A01744_0377_AH27JKDSXF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1277545</td>\n",
       "      <td>A127754501</td>\n",
       "      <td>141-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>19.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/24/25</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>2/11/25</td>\n",
       "      <td>250126_A01021_0847_BH272FDSXF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID  Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "2  XTX101-0102-001                NaN    A1243649  A124364901    101-027   \n",
       "3  XTX101-0102-001                NaN    A1247604  A124760401    146-005   \n",
       "5  XTX101-0102-001                NaN    A1263902  A126390201    141-006   \n",
       "4  XTX101-0102-001                NaN    A1276510  A127651001    140-008   \n",
       "0  XTX101-0102-001                NaN    A1277545  A127754501    141-001   \n",
       "\n",
       "                  Cancertype  \\\n",
       "2            Cecum carcinoma   \n",
       "3  Colorectal adenocarcinoma   \n",
       "5  Colorectal adenocarcinoma   \n",
       "4  Colorectal adenocarcinoma   \n",
       "0  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "2  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "3                                              Stage III         NaN   \n",
       "5  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "4                                              Stage III         NaN   \n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "2       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "3       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "5       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "4       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "0       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "2                        0           21.39              NaN   \n",
       "3                        0           26.64              NaN   \n",
       "5                        0           11.10              NaN   \n",
       "4                        0           33.94              NaN   \n",
       "0                        0           19.86              NaN   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "2                  NaN      12/18/24       12/17/24        1/2/25   \n",
       "3                  NaN      12/21/24       12/19/24        1/2/25   \n",
       "5                  NaN       1/11/25         1/9/25        2/4/25   \n",
       "4                  NaN       1/23/25        1/22/25       2/11/25   \n",
       "0                  NaN       1/24/25        1/23/25       2/11/25   \n",
       "\n",
       "                           runid  \n",
       "2  241221_A01020_0945_AHWTCMDSXC  \n",
       "3  241227_A02019_0258_BHWTKCDSXC  \n",
       "5  250114_A01021_0837_AH27KNDSXF  \n",
       "4  250125_A01744_0377_AH27JKDSXF  \n",
       "0  250126_A01021_0847_BH272FDSXF  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pull \"runid\" from ghdb\n",
    "df_1 = df_dedup.copy()\n",
    "\n",
    "# Define the IDs you want to query for\n",
    "ids = tuple(df_1['GHSampleID'].unique())\n",
    "\n",
    "# Generate the SQL query string\n",
    "query = \"SELECT * FROM gh_sample WHERE run_sample_id IN {}\".format(ids)\n",
    "\n",
    "# Get the dataframe with \"runid\"\n",
    "sample_df = functions.run_sql_query(query)\n",
    "sample_df = sample_df.rename(columns={\"run_sample_id\": \"GHSampleID\"})\n",
    "\n",
    "# Found some sample has duplicated runid which might due to re-run. So only the latest FC runs were saved.\n",
    "# Extract the date from runid and convert to datetime format for accurate comparison\n",
    "sample_df['run_date'] = pd.to_datetime(sample_df['runid'].str.extract(r'(\\d{6})')[0], format='%y%m%d')\n",
    "df_1 = df_1.merge(sample_df[['GHSampleID', 'runid', 'run_date']], on='GHSampleID', how=\"left\")\n",
    "df_1 = df_1.sort_values(by=['GHSampleID', 'run_date'], ascending=[True, False]).drop_duplicates(subset='GHSampleID', keep='first')\n",
    "df_1 = df_1.drop(columns='run_date')\n",
    "\n",
    "print(\"Total sample size of the Infinity batch after merging runid is \" + str(df_1.shape[0])) \n",
    "df_1.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch after merging runid is 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Alt-NoAlt</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>Indel_type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Mut_aa</th>\n",
       "      <th>Mut_nt</th>\n",
       "      <th>Mut_cdna</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>Percentage</th>\n",
       "      <th>Splice_effect</th>\n",
       "      <th>Somatic_status</th>\n",
       "      <th>Molecular_consequence</th>\n",
       "      <th>Fusion_chrom_b</th>\n",
       "      <th>Fusion_gene_b</th>\n",
       "      <th>Fusion_position_a</th>\n",
       "      <th>Fusion_position_b</th>\n",
       "      <th>Direction_a</th>\n",
       "      <th>Direction_b</th>\n",
       "      <th>Downstream_gene</th>\n",
       "      <th>Copy_number</th>\n",
       "      <th>CNV_type</th>\n",
       "      <th>COSMIC</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVarID</th>\n",
       "      <th>Functional_impact</th>\n",
       "      <th>Mutant_allele_status</th>\n",
       "      <th>Mol_count</th>\n",
       "      <th>Max_percentage</th>\n",
       "      <th>Alleletype</th>\n",
       "      <th>HRD_score</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>TMB_score</th>\n",
       "      <th>TMB_category</th>\n",
       "      <th>MSI_High</th>\n",
       "      <th>cfDNA_ng</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Practice_name</th>\n",
       "      <th>Physician_name</th>\n",
       "      <th>runid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C4221015</td>\n",
       "      <td>13431002_Cycle 1 Day 1</td>\n",
       "      <td>A1218829</td>\n",
       "      <td>B00888193</td>\n",
       "      <td>13431002</td>\n",
       "      <td>Cycle 1 Day 1</td>\n",
       "      <td>1</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PIN1</td>\n",
       "      <td>19</td>\n",
       "      <td>9958804.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>A124S</td>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>c.370G&gt;T</td>\n",
       "      <td>NM_006221.4</td>\n",
       "      <td>51.12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>germline</td>\n",
       "      <td>missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs138056875</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5935.0</td>\n",
       "      <td>5.85</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.060326</td>\n",
       "      <td>Detected</td>\n",
       "      <td>0.056315</td>\n",
       "      <td>22.78</td>\n",
       "      <td>High</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>22.85</td>\n",
       "      <td>3.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11/16/2024</td>\n",
       "      <td>06/20/2024</td>\n",
       "      <td>01/17/2025</td>\n",
       "      <td>CRC</td>\n",
       "      <td>Not Applicable</td>\n",
       "      <td>Not Applicable</td>\n",
       "      <td>250106_A00277_1208_BHWTHFDSXC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Study_ID       Customer_SampleID GHRequestID GHSampleID  Patient_ID  \\\n",
       "0  C4221015  13431002_Cycle 1 Day 1    A1218829  B00888193    13431002   \n",
       "\n",
       "      Visit_name  Alt-NoAlt Sample_status  Sample_comment Variant_type  \\\n",
       "0  Cycle 1 Day 1          1       SUCCESS             NaN          SNV   \n",
       "\n",
       "  Indel_type  Gene Chromosome   Position  Exon Mut_aa Mut_nt  Mut_cdna  \\\n",
       "0        NaN  PIN1         19  9958804.0   3.0  A124S    G>T  c.370G>T   \n",
       "\n",
       "    Transcript  Percentage Splice_effect Somatic_status Molecular_consequence  \\\n",
       "0  NM_006221.4       51.12           NaN       germline              missense   \n",
       "\n",
       "   Fusion_chrom_b  Fusion_gene_b  Fusion_position_a  Fusion_position_b  \\\n",
       "0             NaN            NaN                NaN                NaN   \n",
       "\n",
       "   Direction_a  Direction_b  Downstream_gene  Copy_number  CNV_type COSMIC  \\\n",
       "0          NaN          NaN              NaN          NaN       NaN    NaN   \n",
       "\n",
       "         dbSNP ClinVar  ClinVarID Functional_impact  Mutant_allele_status  \\\n",
       "0  rs138056875     NaN        NaN               NaN                   NaN   \n",
       "\n",
       "   Mol_count  Max_percentage  Alleletype  HRD_score Tumor_methylation_status  \\\n",
       "0     5935.0            5.85         NaN   0.060326                 Detected   \n",
       "\n",
       "   Tumor_methylation_score  TMB_score TMB_category      MSI_High  cfDNA_ng  \\\n",
       "0                 0.056315      22.78         High  Not Detected     22.85   \n",
       "\n",
       "   Plasma_ml_input  Plasma_ml_remaining Received_date Bloodcoll_date  \\\n",
       "0              3.5                  0.0    11/16/2024     06/20/2024   \n",
       "\n",
       "  Reported_date Cancertype   Practice_name  Physician_name  \\\n",
       "0    01/17/2025        CRC  Not Applicable  Not Applicable   \n",
       "\n",
       "                           runid  \n",
       "0  250106_A00277_1208_BHWTHFDSXC  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Optional step!!!\n",
    "# # If you only have one sample in the batch, run this cell alternatively instead of from above one\n",
    "\n",
    "# # Pull \"runid\" from ghdb\n",
    "# df_1 = df_dedup.copy()\n",
    "\n",
    "# # Define the IDs you want to query for\n",
    "# ids = tuple(df_1['GHSampleID'].unique())\n",
    "\n",
    "# # If there's only one ID, format it without a comma\n",
    "# if len(ids) == 1:\n",
    "#     ids = f\"('{ids[0]}')\"\n",
    "# else:\n",
    "#     ids = str(ids)\n",
    "\n",
    "# # Generate the SQL query string\n",
    "# query = f\"SELECT * FROM gh_sample WHERE run_sample_id IN {ids}\"\n",
    "\n",
    "# # Get the dataframe with \"runid\"\n",
    "# sample_df = functions.run_sql_query(query)\n",
    "# sample_df = sample_df.rename(columns={\"run_sample_id\": \"GHSampleID\"})\n",
    "\n",
    "# # Found some sample has duplicated runid which might due to re-run. So only the latest FC runs were saved.\n",
    "# # Extract the date from runid and convert to datetime format for accurate comparison\n",
    "# sample_df['run_date'] = pd.to_datetime(sample_df['runid'].str.extract(r'(\\d{6})')[0], format='%y%m%d')\n",
    "# df_1 = df_1.merge(sample_df[['GHSampleID', 'runid', 'run_date']], on='GHSampleID', how=\"left\")\n",
    "# df_1 = df_1.sort_values(by=['GHSampleID', 'run_date'], ascending=[True, False]).drop_duplicates(subset='GHSampleID', keep='first')\n",
    "# df_1 = df_1.drop(columns='run_date')\n",
    "\n",
    "# print(\"Total sample size of the Infinity batch after merging runid is \" + str(df_1.shape[0]))\n",
    "# df_1.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of the Infinity batch with fc_dir is 7\n",
      "\n",
      "Total sample size of the Infinity batch with NO fc_dir is 0\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "      <th>fc_dir</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1243649</td>\n",
       "      <td>A124364901</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>21.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/18/24</td>\n",
       "      <td>12/17/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241221_A01020_0945_AHWTCMDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241221_A01020_0945_AHWTCMDSXC/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1247604</td>\n",
       "      <td>A124760401</td>\n",
       "      <td>146-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>26.64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/21/24</td>\n",
       "      <td>12/19/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241227_A02019_0258_BHWTKCDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241227_A02019_0258_BHWTKCDSXC/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1263902</td>\n",
       "      <td>A126390201</td>\n",
       "      <td>141-006</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>11.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/11/25</td>\n",
       "      <td>1/9/25</td>\n",
       "      <td>2/4/25</td>\n",
       "      <td>250114_A01021_0837_AH27KNDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250114_A01021_0837_AH27KNDSXF/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1276510</td>\n",
       "      <td>A127651001</td>\n",
       "      <td>140-008</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>33.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>1/22/25</td>\n",
       "      <td>2/11/25</td>\n",
       "      <td>250125_A01744_0377_AH27JKDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250125_A01744_0377_AH27JKDSXF/sirius</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1277545</td>\n",
       "      <td>A127754501</td>\n",
       "      <td>141-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>19.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/24/25</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>2/11/25</td>\n",
       "      <td>250126_A01021_0847_BH272FDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250126_A01021_0847_BH272FDSXF.0e716317-4529-4481-89c6-10be60fc1006.20250128124242/sirius</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID  Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "2  XTX101-0102-001                NaN    A1243649  A124364901    101-027   \n",
       "3  XTX101-0102-001                NaN    A1247604  A124760401    146-005   \n",
       "5  XTX101-0102-001                NaN    A1263902  A126390201    141-006   \n",
       "4  XTX101-0102-001                NaN    A1276510  A127651001    140-008   \n",
       "0  XTX101-0102-001                NaN    A1277545  A127754501    141-001   \n",
       "\n",
       "                  Cancertype  \\\n",
       "2            Cecum carcinoma   \n",
       "3  Colorectal adenocarcinoma   \n",
       "5  Colorectal adenocarcinoma   \n",
       "4  Colorectal adenocarcinoma   \n",
       "0  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "2  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "3                                              Stage III         NaN   \n",
       "5  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "4                                              Stage III         NaN   \n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "2       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "3       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "5       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "4       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "0       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "2                        0           21.39              NaN   \n",
       "3                        0           26.64              NaN   \n",
       "5                        0           11.10              NaN   \n",
       "4                        0           33.94              NaN   \n",
       "0                        0           19.86              NaN   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "2                  NaN      12/18/24       12/17/24        1/2/25   \n",
       "3                  NaN      12/21/24       12/19/24        1/2/25   \n",
       "5                  NaN       1/11/25         1/9/25        2/4/25   \n",
       "4                  NaN       1/23/25        1/22/25       2/11/25   \n",
       "0                  NaN       1/24/25        1/23/25       2/11/25   \n",
       "\n",
       "                           runid  \\\n",
       "2  241221_A01020_0945_AHWTCMDSXC   \n",
       "3  241227_A02019_0258_BHWTKCDSXC   \n",
       "5  250114_A01021_0837_AH27KNDSXF   \n",
       "4  250125_A01744_0377_AH27JKDSXF   \n",
       "0  250126_A01021_0847_BH272FDSXF   \n",
       "\n",
       "                                                                                                            fc_dir  \n",
       "2                                                      /ghsfa/ivd/flowcentral/241221_A01020_0945_AHWTCMDSXC/sirius  \n",
       "3                                                      /ghsfa/ivd/flowcentral/241227_A02019_0258_BHWTKCDSXC/sirius  \n",
       "5                                                      /ghsfa/ivd/flowcentral/250114_A01021_0837_AH27KNDSXF/sirius  \n",
       "4                                                      /ghsfa/ivd/flowcentral/250125_A01744_0377_AH27JKDSXF/sirius  \n",
       "0  /ghsfa/ivd/flowcentral/250126_A01021_0847_BH272FDSXF.0e716317-4529-4481-89c6-10be60fc1006.20250128124242/sirius  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def add_fc_dir_column(df):\n",
    "    \"\"\"\n",
    "    Adds a 'fc_dir' column to the given dataframe, specifying the FC path based on the 'runid' column.\n",
    "\n",
    "    Args:\n",
    "        df (pd.DataFrame): A DataFrame that must contain a 'runid' column. This column is used to\n",
    "                           match directory paths from predefined locations.\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The original dataframe with an added 'fc_dir' column indicating the directory path\n",
    "                      corresponding to each 'runid'. If no directory is found, the path will be an empty string.\n",
    "        pd.DataFrame: A DataFrame containing the rows for which no directory path was found.\n",
    "    \"\"\"\n",
    "    # Define the path\n",
    "    fc_dir_all = \"/ghsfa/ivd/flowcentral\"\n",
    "    fc_dir_omni = \"/ghsfa/projects/omni_v2/flowcells\"\n",
    "    \n",
    "    # Initialize a list to store rows where no BIP output was found\n",
    "    rows_no_found = []\n",
    "    \n",
    "    # Initialize the new column in the DataFrame\n",
    "    df['fc_dir'] = ''\n",
    "    \n",
    "    # Iterate over the DataFrame rows\n",
    "    for index, row in df.iterrows():\n",
    "        runid = row[\"runid\"]\n",
    "        \n",
    "        # Find original folder\n",
    "        pattern_ori = f\"{fc_dir_all}/*{runid}*\"\n",
    "        folders = glob.glob(pattern_ori)\n",
    "        if folders:\n",
    "            for folder in folders:\n",
    "                primary_fc_dir = folder\n",
    "                # Check if there is a 'sirius' folder under the primary fc_dir. This is for Sirius 1.5 folder structure change.\n",
    "                if os.path.exists(os.path.join(primary_fc_dir, \"sirius\")):\n",
    "                    primary_fc_dir = os.path.join(primary_fc_dir, \"sirius\")\n",
    "                df.loc[index, 'fc_dir'] = primary_fc_dir\n",
    "        else:\n",
    "            # Search in the alternative directory if no folder found in the first directory\n",
    "            pattern_omni = f\"{fc_dir_omni}/*{runid}*\"\n",
    "            folders_omni = glob.glob(pattern_omni)\n",
    "            if folders_omni:\n",
    "                df.loc[index, 'fc_dir'] = folders_omni[0]\n",
    "            else:\n",
    "                # Append the row to the list if no folder is found in either location\n",
    "                rows_no_found.append(row)\n",
    "    \n",
    "    # Create a DataFrame with rows where folders were not found in any sources\n",
    "    df_no_fc_dir = pd.DataFrame(rows_no_found)\n",
    "    \n",
    "    return df, df_no_fc_dir\n",
    "\n",
    "# Generate the new dataframe \n",
    "df_2, df_no_found = add_fc_dir_column(df_1)\n",
    "\n",
    "print(\"Total sample size of the Infinity batch with fc_dir is \" + str(df_2.shape[0]) + \"\\n\") \n",
    "print(\"Total sample size of the Infinity batch with NO fc_dir is \" + str(df_no_found.shape[0]) + \"\\n\") \n",
    "\n",
    "df_2.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate S3 MB caller outputs using Sirius 1.6.x yaml\n",
    "###### Git clone \"Sirius-1.6.0_Release_Branch\" and \"devel/mcai/update_crc_model_file\" (Sirius-1.6.1) of bip-products for corresponding CRC caller cutoffs and correct CRC region_file for MB CRC caller\n",
    "###### bip-epigenomics module instead of epi-caller was used for CRC result comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Instead of using epicaller.methylation.methyl_caller funciton, bip-epigenomics module (branch 2.5) is prefered.\n",
    "\n",
    "def run_bip_epigenomics_module_wrapper_slurm_Sirius161(sample_meta_df, workdir):\n",
    "    module_dir = '/home/mcai/workspace/bip-epigenomics-2.5.3-RLS/bip-epigenomics/'  # branch bip-epigenomics-2.5.3-RLS for sirius 1.6.1\n",
    "    parameter_set_yaml = \"/ghsfa/projects/pharma/customers/GuardantHealthIST/Infinity/MDC04_IST01_Infinity/MDC04_IST01_Infinity_01/data/rerun_crc_caller/bip-products/Sirius_LDT_Soft/Sirius_LDT_Soft.yaml\" # This is for Sirius 1.6.1\n",
    "\n",
    "    # Directories for jobs and output \n",
    "    jobs_dir = os.path.join(workdir, \"MB_module_CRC_jobs\")\n",
    "    output_base_dir = os.path.join(workdir, \"MB_module_CRC_output\")\n",
    "    \n",
    "    # Create directories if they don't exist\n",
    "    os.makedirs(jobs_dir, exist_ok=True)\n",
    "    os.makedirs(output_base_dir, exist_ok=True)\n",
    "    \n",
    "    for _, row in sample_meta_df.iterrows():\n",
    "        run_sample_id = row[\"GHSampleID\"]\n",
    "        fc_dir = row[\"fc_dir\"]\n",
    "        sample_dir = f\"{fc_dir}/{run_sample_id}/\"\n",
    "        \n",
    "        # Define input files from sample_dir\n",
    "        mbd_hyper_molecules = os.path.join(sample_dir, f\"{run_sample_id}.mbd_hyper_molecules.tsv\")\n",
    "        methyl_qc_file = os.path.join(sample_dir, f\"{run_sample_id}.methyl_qc.tsv\")\n",
    "        subpanel_qc_file = os.path.join(sample_dir, f\"{run_sample_id}.subpanel_qc_stats.hdr.tsv\")\n",
    "        \n",
    "        # Skip sample if the main input file is missing\n",
    "        if not os.path.exists(mbd_hyper_molecules):\n",
    "            continue\n",
    "        \n",
    "        # Create output directory for the sample (mimics out_path/sampleID)\n",
    "        out_dir = os.path.join(output_base_dir, run_sample_id)\n",
    "        os.makedirs(out_dir, exist_ok=True)\n",
    "        \n",
    "        # Copy bip_config.json from fc_dir to the sample's output directory\n",
    "        bip_config_json_file = os.path.join(fc_dir, \"bip_config.json\")\n",
    "        os.system(f\"cp {bip_config_json_file} {out_dir}\")\n",
    "        \n",
    "        # Create a job file that will run all three steps sequentially\n",
    "        job_file = os.path.join(jobs_dir, f\"MB_call_{run_sample_id}.job\")\n",
    "        \n",
    "        # Refined SLURM header\n",
    "        shebang_content = (\n",
    "            \"#!/bin/bash\\n\"\n",
    "            f\"#SBATCH --job-name=MB_call_{run_sample_id}\\n\"\n",
    "            \"#SBATCH --nodes=1\\n\"\n",
    "            \"#SBATCH --ntasks=1\\n\"\n",
    "            \"#SBATCH --cpus-per-task=16\\n\"\n",
    "            \"#SBATCH --mem=150g\\n\"\n",
    "            \"#SBATCH --partition=dev\\n\"\n",
    "            f\"#SBATCH --output={jobs_dir}/MB_call_{run_sample_id}.log\\n\"\n",
    "            f\"#SBATCH --error={jobs_dir}/MB_call_{run_sample_id}.err\\n\\n\"\n",
    "        )\n",
    "        \n",
    "        # 1. run msre_caller\n",
    "        cmd_msre_caller = (\n",
    "            f\"{module_dir}/module_entry msre_caller \"\n",
    "            f\"--in-hyper-molecules {mbd_hyper_molecules} \"\n",
    "            f\"--sample-id {run_sample_id} \"\n",
    "            f\"--output-dir {out_dir} \"\n",
    "            f\"--parameter-set {parameter_set_yaml} \"\n",
    "            f\"--threads 16 \"\n",
    "            f\"--memory-log {out_dir}/{run_sample_id}-epigenomics_msre_caller.memory.log \"\n",
    "            f\"--log-file {out_dir}/{run_sample_id}-epigenomics_msre_caller.log\\n\"\n",
    "        )\n",
    "        \n",
    "        # 2. run epigenomics_tumor_fraction\n",
    "        cmd_tumor_fraction = (\n",
    "            f\"{module_dir}/module_entry epigenomics_tumor_fraction \"\n",
    "            f\"--in-hyper-molecules {mbd_hyper_molecules} \"\n",
    "            f\"--in-methyl-qc-file {methyl_qc_file} \"\n",
    "            f\"--in-subpanel-qc-file {subpanel_qc_file} \"\n",
    "            f\"--sample-id {run_sample_id} \"\n",
    "            f\"--output-dir {out_dir} \"\n",
    "            f\"--parameter-set {parameter_set_yaml} \"\n",
    "            f\"--threads 2 \"\n",
    "            f\"--memory-log {out_dir}/{run_sample_id}-epigenomics_tumor_fraction.memory.log \"\n",
    "            f\"--log-file {out_dir}/{run_sample_id}-epigenomics_tumor_fraction.log\\n\"\n",
    "        )\n",
    "        \n",
    "        # 3. run collect_methyl_callers\n",
    "        cmd_collect = (\n",
    "            f\"{module_dir}/module_entry collect_methyl_callers \"\n",
    "            f\"--in-msre-crc-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_crc_call.hdr.tsv \"\n",
    "            f\"--in-msre-lung-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_lung_call.hdr.tsv \"\n",
    "            f\"--in-msre-breast-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_breast_call.hdr.tsv \"\n",
    "            f\"--in-msre-pancancer-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_pancancer_call.hdr.tsv \"\n",
    "            f\"--in-tf-estimate {out_dir}/{run_sample_id}/{run_sample_id}.epi_tf_model_predictions.tsv \"\n",
    "            f\"--sample-id {run_sample_id} \"\n",
    "            f\"--output {out_dir} \"\n",
    "            f\"--parameter-set {parameter_set_yaml} \"\n",
    "            f\"--threads 2 \"\n",
    "            f\"--memory-log {out_dir}/{run_sample_id}-epigenomics_msre_summary.memory.log \"\n",
    "            f\"--log-file {out_dir}/{run_sample_id}-epigenomics_msre_summary.log\\n\"\n",
    "        )\n",
    "        \n",
    "        # Write all commands into the job file\n",
    "        with open(job_file, \"w\") as jf:\n",
    "            jf.write(shebang_content)\n",
    "            jf.write(cmd_msre_caller)\n",
    "            jf.write(cmd_tumor_fraction)\n",
    "            jf.write(cmd_collect)\n",
    "        \n",
    "        # Submit the job file via sbatch\n",
    "        os.system(f\"sbatch {job_file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Instead of using epicaller.methylation.methyl_caller funciton, bip-epigenomics module (branch 2.5) is prefered.\n",
    "\n",
    "def run_bip_epigenomics_module_wrapper_slurm_Sirius160(sample_meta_df, workdir):\n",
    "    module_dir = '/home/mcai/workspace/bip-epigenomics/'  # branch bip-epigenomics-2.5.0-RC2 for sirius 1.6.0\n",
    "    parameter_set_yaml = \"/ghdevhome/home/hazhang/projects/Repo/bip-products_versions/bip-products-Sirius160/bip-products/Sirius_LDT_Soft/Sirius_LDT_Soft.yaml\" # This is for Sirius 1.6.0\n",
    "\n",
    "    # Directories for jobs and output \n",
    "    jobs_dir = os.path.join(workdir, \"MB_module_CRC_jobs\")\n",
    "    output_base_dir = os.path.join(workdir, \"MB_module_CRC_output\")\n",
    "    \n",
    "    # Create directories if they don't exist\n",
    "    os.makedirs(jobs_dir, exist_ok=True)\n",
    "    os.makedirs(output_base_dir, exist_ok=True)\n",
    "    \n",
    "    for _, row in sample_meta_df.iterrows():\n",
    "        run_sample_id = row[\"GHSampleID\"]\n",
    "        fc_dir = row[\"fc_dir\"]\n",
    "        sample_dir = f\"{fc_dir}/{run_sample_id}/\"\n",
    "        \n",
    "        # Define input files from sample_dir\n",
    "        mbd_hyper_molecules = os.path.join(sample_dir, f\"{run_sample_id}.mbd_hyper_molecules.tsv\")\n",
    "        methyl_qc_file = os.path.join(sample_dir, f\"{run_sample_id}.methyl_qc.tsv\")\n",
    "        subpanel_qc_file = os.path.join(sample_dir, f\"{run_sample_id}.subpanel_qc_stats.hdr.tsv\")\n",
    "        \n",
    "        # Skip sample if the main input file is missing\n",
    "        if not os.path.exists(mbd_hyper_molecules):\n",
    "            continue\n",
    "        \n",
    "        # Create output directory for the sample (mimics out_path/sampleID)\n",
    "        out_dir = os.path.join(output_base_dir, run_sample_id)\n",
    "        os.makedirs(out_dir, exist_ok=True)\n",
    "        \n",
    "        # Copy bip_config.json from fc_dir to the sample's output directory\n",
    "        bip_config_json_file = os.path.join(fc_dir, \"bip_config.json\")\n",
    "        os.system(f\"cp {bip_config_json_file} {out_dir}\")\n",
    "        \n",
    "        # Create a job file that will run all three steps sequentially\n",
    "        job_file = os.path.join(jobs_dir, f\"MB_call_{run_sample_id}.job\")\n",
    "        \n",
    "        # Refined SLURM header\n",
    "        shebang_content = (\n",
    "            \"#!/bin/bash\\n\"\n",
    "            f\"#SBATCH --job-name=MB_call_{run_sample_id}\\n\"\n",
    "            \"#SBATCH --nodes=1\\n\"\n",
    "            \"#SBATCH --ntasks=1\\n\"\n",
    "            \"#SBATCH --cpus-per-task=16\\n\"\n",
    "            \"#SBATCH --mem=150g\\n\"\n",
    "            \"#SBATCH --partition=dev\\n\"\n",
    "            f\"#SBATCH --output={jobs_dir}/MB_call_{run_sample_id}.log\\n\"\n",
    "            f\"#SBATCH --error={jobs_dir}/MB_call_{run_sample_id}.err\\n\\n\"\n",
    "        )\n",
    "        \n",
    "        # 1. run msre_caller\n",
    "        cmd_msre_caller = (\n",
    "            f\"{module_dir}/module_entry msre_caller \"\n",
    "            f\"--in-hyper-molecules {mbd_hyper_molecules} \"\n",
    "            f\"--sample-id {run_sample_id} \"\n",
    "            f\"--output-dir {out_dir} \"\n",
    "            f\"--parameter-set {parameter_set_yaml} \"\n",
    "            f\"--threads 16 \"\n",
    "            f\"--memory-log {out_dir}/{run_sample_id}-epigenomics_msre_caller.memory.log \"\n",
    "            f\"--log-file {out_dir}/{run_sample_id}-epigenomics_msre_caller.log\\n\"\n",
    "        )\n",
    "        \n",
    "        # 2. run epigenomics_tumor_fraction\n",
    "        cmd_tumor_fraction = (\n",
    "            f\"{module_dir}/module_entry epigenomics_tumor_fraction \"\n",
    "            f\"--in-hyper-molecules {mbd_hyper_molecules} \"\n",
    "            f\"--in-methyl-qc-file {methyl_qc_file} \"\n",
    "            f\"--in-subpanel-qc-file {subpanel_qc_file} \"\n",
    "            f\"--sample-id {run_sample_id} \"\n",
    "            f\"--output-dir {out_dir} \"\n",
    "            f\"--parameter-set {parameter_set_yaml} \"\n",
    "            f\"--threads 2 \"\n",
    "            f\"--memory-log {out_dir}/{run_sample_id}-epigenomics_tumor_fraction.memory.log \"\n",
    "            f\"--log-file {out_dir}/{run_sample_id}-epigenomics_tumor_fraction.log\\n\"\n",
    "        )\n",
    "        \n",
    "        # 3. run collect_methyl_callers\n",
    "        cmd_collect = (\n",
    "            f\"{module_dir}/module_entry collect_methyl_callers \"\n",
    "            f\"--in-msre-crc-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_crc_call.hdr.tsv \"\n",
    "            f\"--in-msre-lung-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_lung_call.hdr.tsv \"\n",
    "            f\"--in-msre-breast-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_breast_call.hdr.tsv \"\n",
    "            f\"--in-msre-pancancer-call {out_dir}/{run_sample_id}/{run_sample_id}.msre_pancancer_call.hdr.tsv \"\n",
    "            f\"--in-tf-estimate {out_dir}/{run_sample_id}/{run_sample_id}.epi_tf_model_predictions.tsv \"\n",
    "            f\"--sample-id {run_sample_id} \"\n",
    "            f\"--output {out_dir} \"\n",
    "            f\"--parameter-set {parameter_set_yaml} \"\n",
    "            f\"--threads 2 \"\n",
    "            f\"--memory-log {out_dir}/{run_sample_id}-epigenomics_msre_summary.memory.log \"\n",
    "            f\"--log-file {out_dir}/{run_sample_id}-epigenomics_msre_summary.log\\n\"\n",
    "        )\n",
    "        \n",
    "        # Write all commands into the job file\n",
    "        with open(job_file, \"w\") as jf:\n",
    "            jf.write(shebang_content)\n",
    "            jf.write(cmd_msre_caller)\n",
    "            jf.write(cmd_tumor_fraction)\n",
    "            jf.write(cmd_collect)\n",
    "        \n",
    "        # Submit the job file via sbatch\n",
    "        os.system(f\"sbatch {job_file}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 101496965\n",
      "Submitted batch job 101496966\n",
      "Submitted batch job 101496967\n",
      "Submitted batch job 101496968\n",
      "Submitted batch job 101496969\n",
      "Submitted batch job 101496970\n",
      "Submitted batch job 101496971\n"
     ]
    }
   ],
   "source": [
    "#run the function to generate all the MB S3 caller outputs\n",
    "run_bip_epigenomics_module_wrapper_slurm_Sirius161(df_2, f\"{work_dir}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sample size of samples with NO MB output is 0\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "      <th>fc_dir</th>\n",
       "      <th>MB_CRC_call_Sirius150</th>\n",
       "      <th>MB_CRC_score_Sirius150</th>\n",
       "      <th>pred_frac</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1243649</td>\n",
       "      <td>A124364901</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>21.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/18/24</td>\n",
       "      <td>12/17/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241221_A01020_0945_AHWTCMDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241221_A01020_0945_AHWTCMDSXC/sirius</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>-21.906225</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1247604</td>\n",
       "      <td>A124760401</td>\n",
       "      <td>146-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>26.64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/21/24</td>\n",
       "      <td>12/19/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241227_A02019_0258_BHWTKCDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241227_A02019_0258_BHWTKCDSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-20.906050</td>\n",
       "      <td>0.000270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1263902</td>\n",
       "      <td>A126390201</td>\n",
       "      <td>141-006</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>11.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/11/25</td>\n",
       "      <td>1/9/25</td>\n",
       "      <td>2/4/25</td>\n",
       "      <td>250114_A01021_0837_AH27KNDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250114_A01021_0837_AH27KNDSXF/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-20.515209</td>\n",
       "      <td>0.000103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1276510</td>\n",
       "      <td>A127651001</td>\n",
       "      <td>140-008</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>33.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>1/22/25</td>\n",
       "      <td>2/11/25</td>\n",
       "      <td>250125_A01744_0377_AH27JKDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250125_A01744_0377_AH27JKDSXF/sirius</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>-21.774074</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1277545</td>\n",
       "      <td>A127754501</td>\n",
       "      <td>141-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>19.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/24/25</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>2/11/25</td>\n",
       "      <td>250126_A01021_0847_BH272FDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250126_A01021_0847_BH272FDSXF.0e716317-4529-4481-89c6-10be60fc1006.20250128124242/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-20.986001</td>\n",
       "      <td>0.000235</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID  Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "2  XTX101-0102-001                NaN    A1243649  A124364901    101-027   \n",
       "3  XTX101-0102-001                NaN    A1247604  A124760401    146-005   \n",
       "5  XTX101-0102-001                NaN    A1263902  A126390201    141-006   \n",
       "4  XTX101-0102-001                NaN    A1276510  A127651001    140-008   \n",
       "0  XTX101-0102-001                NaN    A1277545  A127754501    141-001   \n",
       "\n",
       "                  Cancertype  \\\n",
       "2            Cecum carcinoma   \n",
       "3  Colorectal adenocarcinoma   \n",
       "5  Colorectal adenocarcinoma   \n",
       "4  Colorectal adenocarcinoma   \n",
       "0  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "2  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "3                                              Stage III         NaN   \n",
       "5  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "4                                              Stage III         NaN   \n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "2       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "3       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "5       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "4       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "0       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "2                        0           21.39              NaN   \n",
       "3                        0           26.64              NaN   \n",
       "5                        0           11.10              NaN   \n",
       "4                        0           33.94              NaN   \n",
       "0                        0           19.86              NaN   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "2                  NaN      12/18/24       12/17/24        1/2/25   \n",
       "3                  NaN      12/21/24       12/19/24        1/2/25   \n",
       "5                  NaN       1/11/25         1/9/25        2/4/25   \n",
       "4                  NaN       1/23/25        1/22/25       2/11/25   \n",
       "0                  NaN       1/24/25        1/23/25       2/11/25   \n",
       "\n",
       "                           runid  \\\n",
       "2  241221_A01020_0945_AHWTCMDSXC   \n",
       "3  241227_A02019_0258_BHWTKCDSXC   \n",
       "5  250114_A01021_0837_AH27KNDSXF   \n",
       "4  250125_A01744_0377_AH27JKDSXF   \n",
       "0  250126_A01021_0847_BH272FDSXF   \n",
       "\n",
       "                                                                                                            fc_dir  \\\n",
       "2                                                      /ghsfa/ivd/flowcentral/241221_A01020_0945_AHWTCMDSXC/sirius   \n",
       "3                                                      /ghsfa/ivd/flowcentral/241227_A02019_0258_BHWTKCDSXC/sirius   \n",
       "5                                                      /ghsfa/ivd/flowcentral/250114_A01021_0837_AH27KNDSXF/sirius   \n",
       "4                                                      /ghsfa/ivd/flowcentral/250125_A01744_0377_AH27JKDSXF/sirius   \n",
       "0  /ghsfa/ivd/flowcentral/250126_A01021_0847_BH272FDSXF.0e716317-4529-4481-89c6-10be60fc1006.20250128124242/sirius   \n",
       "\n",
       "  MB_CRC_call_Sirius150  MB_CRC_score_Sirius150  pred_frac  \n",
       "2          Not Detected              -21.906225   0.000000  \n",
       "3              Detected              -20.906050   0.000270  \n",
       "5              Detected              -20.515209   0.000103  \n",
       "4          Not Detected              -21.774074   0.000000  \n",
       "0              Detected              -20.986001   0.000235  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Retrieve MB S3 CRC caller score and MB S3 Lung caller score based on the cancer type and stage\n",
    "df_3 = df_2.copy()\n",
    "\n",
    "df_3[\"MB_CRC_call_Sirius150\"] = np.nan\n",
    "df_3[\"MB_CRC_score_Sirius150\"] = np.nan\n",
    "df_3[\"pred_frac\"] = np.nan\n",
    "\n",
    "mb_caller_result_folder = f\"{work_dir}/MB_module_CRC_output\"\n",
    "problematic_rows = []  # List to store indices of problematic rows\n",
    "\n",
    "for index, crc_row in df_3.iterrows():\n",
    "    run_sample_id = crc_row[\"GHSampleID\"]\n",
    "    msre_mr_file = f\"{mb_caller_result_folder}/{run_sample_id}/{run_sample_id}/{run_sample_id}.msre_all_call.hdr.tsv\"   \n",
    "    try:\n",
    "        msre_df = pd.read_table(msre_mr_file)\n",
    "        msre_df_crc = msre_df[msre_df['model_name'] == 'CRC_LateStage']\n",
    "        # print(msre_df_crc)\n",
    "\n",
    "        if not msre_df_crc.empty:\n",
    "            df_3.at[index, \"pred_frac\"] = msre_df_crc.iloc[0][\"pred_frac\"]\n",
    "            df_3.at[index, \"MB_CRC_score_Sirius150\"] = msre_df_crc.iloc[0][\"methyl_score\"]\n",
    "            df_3.at[index, \"MB_CRC_call_Sirius150\"] = \"Detected\" if msre_df_crc.iloc[0][\"methyl_call\"] == 1 else \"Not Detected\"\n",
    "        else:\n",
    "            # Add the index to the list if msre_df is empty\n",
    "            problematic_rows.append(index) \n",
    "            \n",
    "    except Exception as e:\n",
    "        # Add the index to the list if there is any other exception\n",
    "        problematic_rows.append(index)  \n",
    "        print(f\"Error processing file {msre_mr_file}: {str(e)}\")\n",
    "\n",
    "# Create a DataFrame with the indices of the problematic rows\n",
    "df_problematic = df_3.loc[problematic_rows]\n",
    "\n",
    "print(\"Total sample size of samples with NO MB output is \" + str(df_problematic.shape[0]) + \"\\n\") \n",
    "\n",
    "# Returning the DataFrame with problematic rows\n",
    "df_3.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check the result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total CRC samples with flipped call is 5\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study_ID</th>\n",
       "      <th>Customer_SampleID</th>\n",
       "      <th>GHRequestID</th>\n",
       "      <th>GHSampleID</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Cancertype</th>\n",
       "      <th>Tumor_stage</th>\n",
       "      <th>Visit_name</th>\n",
       "      <th>Sample_status</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>ctDNA_detected</th>\n",
       "      <th>Tumor_methylation_status</th>\n",
       "      <th>Tumor_methylation_score</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Plasma_ml_remaining</th>\n",
       "      <th>Received_date</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>Reported_date</th>\n",
       "      <th>runid</th>\n",
       "      <th>fc_dir</th>\n",
       "      <th>MB_CRC_call_Sirius150</th>\n",
       "      <th>MB_CRC_score_Sirius150</th>\n",
       "      <th>pred_frac</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1247604</td>\n",
       "      <td>A124760401</td>\n",
       "      <td>146-005</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>26.64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/21/24</td>\n",
       "      <td>12/19/24</td>\n",
       "      <td>1/2/25</td>\n",
       "      <td>241227_A02019_0258_BHWTKCDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241227_A02019_0258_BHWTKCDSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-20.906050</td>\n",
       "      <td>0.000270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1263902</td>\n",
       "      <td>A126390201</td>\n",
       "      <td>141-006</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>11.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/11/25</td>\n",
       "      <td>1/9/25</td>\n",
       "      <td>2/4/25</td>\n",
       "      <td>250114_A01021_0837_AH27KNDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250114_A01021_0837_AH27KNDSXF/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-20.515209</td>\n",
       "      <td>0.000103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1277545</td>\n",
       "      <td>A127754501</td>\n",
       "      <td>141-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>19.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/24/25</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>2/11/25</td>\n",
       "      <td>250126_A01021_0847_BH272FDSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250126_A01021_0847_BH272FDSXF.0e716317-4529-4481-89c6-10be60fc1006.20250128124242/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-20.986001</td>\n",
       "      <td>0.000235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1285830</td>\n",
       "      <td>A128583001</td>\n",
       "      <td>101-027</td>\n",
       "      <td>Cecum carcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>27.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/31/25</td>\n",
       "      <td>1/30/25</td>\n",
       "      <td>2/19/25</td>\n",
       "      <td>250202_A00569_0940_AH3CM5DSXF</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/250202_A00569_0940_AH3CM5DSXF/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-21.213370</td>\n",
       "      <td>0.000045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>XTX101-0102-001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1218823</td>\n",
       "      <td>B00848429</td>\n",
       "      <td>141-001</td>\n",
       "      <td>Colorectal adenocarcinoma</td>\n",
       "      <td>Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>Not Detected</td>\n",
       "      <td>0</td>\n",
       "      <td>13.21</td>\n",
       "      <td>3.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11/22/24</td>\n",
       "      <td>11/21/24</td>\n",
       "      <td>12/17/24</td>\n",
       "      <td>241204_A01112_0657_BHTFJKDSXC</td>\n",
       "      <td>/ghsfa/ivd/flowcentral/241204_A01112_0657_BHTFJKDSXC/sirius</td>\n",
       "      <td>Detected</td>\n",
       "      <td>-21.248167</td>\n",
       "      <td>0.000099</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Study_ID  Customer_SampleID GHRequestID  GHSampleID Patient_ID  \\\n",
       "3  XTX101-0102-001                NaN    A1247604  A124760401    146-005   \n",
       "5  XTX101-0102-001                NaN    A1263902  A126390201    141-006   \n",
       "0  XTX101-0102-001                NaN    A1277545  A127754501    141-001   \n",
       "1  XTX101-0102-001                NaN    A1285830  A128583001    101-027   \n",
       "6  XTX101-0102-001                NaN    A1218823   B00848429    141-001   \n",
       "\n",
       "                  Cancertype  \\\n",
       "3  Colorectal adenocarcinoma   \n",
       "5  Colorectal adenocarcinoma   \n",
       "0  Colorectal adenocarcinoma   \n",
       "1            Cecum carcinoma   \n",
       "6  Colorectal adenocarcinoma   \n",
       "\n",
       "                                             Tumor_stage  Visit_name  \\\n",
       "3                                              Stage III         NaN   \n",
       "5  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "0  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "1  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "6  Stage IIIB/IV NSCLC, Stage III/IV - Other Cancer Type         NaN   \n",
       "\n",
       "  Sample_status  Sample_comment ctDNA_detected Tumor_methylation_status  \\\n",
       "3       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "5       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "0       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "1       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "6       SUCCESS             NaN   Not Detected             Not Detected   \n",
       "\n",
       "   Tumor_methylation_score  cfDNA_ng_yield  Plasma_ml_input  \\\n",
       "3                        0           26.64              NaN   \n",
       "5                        0           11.10              NaN   \n",
       "0                        0           19.86              NaN   \n",
       "1                        0           27.23              NaN   \n",
       "6                        0           13.21              3.5   \n",
       "\n",
       "   Plasma_ml_remaining Received_date Bloodcoll_date Reported_date  \\\n",
       "3                  NaN      12/21/24       12/19/24        1/2/25   \n",
       "5                  NaN       1/11/25         1/9/25        2/4/25   \n",
       "0                  NaN       1/24/25        1/23/25       2/11/25   \n",
       "1                  NaN       1/31/25        1/30/25       2/19/25   \n",
       "6                  NaN      11/22/24       11/21/24      12/17/24   \n",
       "\n",
       "                           runid  \\\n",
       "3  241227_A02019_0258_BHWTKCDSXC   \n",
       "5  250114_A01021_0837_AH27KNDSXF   \n",
       "0  250126_A01021_0847_BH272FDSXF   \n",
       "1  250202_A00569_0940_AH3CM5DSXF   \n",
       "6  241204_A01112_0657_BHTFJKDSXC   \n",
       "\n",
       "                                                                                                            fc_dir  \\\n",
       "3                                                      /ghsfa/ivd/flowcentral/241227_A02019_0258_BHWTKCDSXC/sirius   \n",
       "5                                                      /ghsfa/ivd/flowcentral/250114_A01021_0837_AH27KNDSXF/sirius   \n",
       "0  /ghsfa/ivd/flowcentral/250126_A01021_0847_BH272FDSXF.0e716317-4529-4481-89c6-10be60fc1006.20250128124242/sirius   \n",
       "1                                                      /ghsfa/ivd/flowcentral/250202_A00569_0940_AH3CM5DSXF/sirius   \n",
       "6                                                      /ghsfa/ivd/flowcentral/241204_A01112_0657_BHTFJKDSXC/sirius   \n",
       "\n",
       "  MB_CRC_call_Sirius150  MB_CRC_score_Sirius150  pred_frac  \n",
       "3              Detected              -20.906050   0.000270  \n",
       "5              Detected              -20.515209   0.000103  \n",
       "0              Detected              -20.986001   0.000235  \n",
       "1              Detected              -21.213370   0.000045  \n",
       "6              Detected              -21.248167   0.000099  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_flipped_call = df_3[df_3['MB_CRC_call_Sirius150'] != df_3['Tumor_methylation_status']]\n",
    "print(\"Total CRC samples with flipped call is \" + str(df_flipped_call.shape[0]) + \"\\n\")\n",
    "\n",
    "# Save the data\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20250121_Pfizer_SOW51core_Batch3_18Samples_InfinityReport_flipped_calls.csv\", index = False)\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20241220_GuardantHealthIST_IST01_Batch1_405Samples_InfinityReport_flipped_calls.csv\", index = False)\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20250120_Pfizer_SOW46Amendment1TP3_Batch9_20Samples_G360Report_flipped_calls.csv\", index = False)\n",
    "# df_flipped_call.to_csv(f\"{work_dir}/20250120_Pfizer_SOW46Amendment1TP5_Batch7_33Samples_InfinityReport_flipped_calls.csv\", index = False)\n",
    "df_flipped_call.to_csv(f\"{work_dir}/XTX_monitoring_Sirius161_flipped_calls_by_cancer_stage_cutoff_03072025.csv\", index = False)\n",
    "\n",
    "df_flipped_call\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "lunartk_v1.1.0",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
