{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7f16768c-98a3-4f25-b9ba-35d754757d2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import subprocess\n",
    "import os\n",
    "import glob\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy\n",
    "import scipy.stats as stats\n",
    "from sklearn.decomposition import PCA\n",
    "import pickle\n",
    "import re\n",
    "import math\n",
    "import plotnine as pn\n",
    "import seaborn as sns\n",
    "import glob\n",
    "import subprocess\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "import seaborn as sns\n",
    "from numpy import log\n",
    "from collections import defaultdict,Counter\n",
    "import re\n",
    "import statsmodels\n",
    "import warnings\n",
    "\n",
    "# !pip install statannotations==0.4.0\n",
    "# #!pip uninstall statannotations --yes\n",
    "# from statannotations.Annotator import Annotator\n",
    "\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "pd.options.display.max_rows = None\n",
    "pd.options.display.max_columns=None\n",
    "pd.options.display.max_colwidth=None\n",
    "work_dir=\"/home/hazhang/projects/VHL_promoter_methylation_12122023\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17f08915-269d-40d1-9111-773cae4146ab",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Data processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e8997e76-86a7-4336-a299-ac21980c7e5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>runid</th>\n",
       "      <th>ctrl_cor_frac</th>\n",
       "      <th>pos_ctrl_after_correction</th>\n",
       "      <th>pred_type</th>\n",
       "      <th>CRC</th>\n",
       "      <th>LUNG</th>\n",
       "      <th>BREAST</th>\n",
       "      <th>BLADDER</th>\n",
       "      <th>AGGREGATED</th>\n",
       "      <th>CRC_TF</th>\n",
       "      <th>LUNG_TF</th>\n",
       "      <th>BREAST_TF</th>\n",
       "      <th>BLADDER_TF</th>\n",
       "      <th>AGGREGATED_TF</th>\n",
       "      <th>cohort</th>\n",
       "      <th>request_id</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>Alt-NoAlt</th>\n",
       "      <th>SUCCESS</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>max_maf_pct_report</th>\n",
       "      <th>RUO_methylation_status</th>\n",
       "      <th>RUO_methylation_tf</th>\n",
       "      <th>TMB_score</th>\n",
       "      <th>TMB_category</th>\n",
       "      <th>MSI_High</th>\n",
       "      <th>input_ng</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>raw_cancer_type</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>subset</th>\n",
       "      <th>genomic_method</th>\n",
       "      <th>fc_name</th>\n",
       "      <th>cram_path</th>\n",
       "      <th>somatic_max_maf_gene</th>\n",
       "      <th>mut_aa</th>\n",
       "      <th>somatic_max_maf_pct</th>\n",
       "      <th>is_titration</th>\n",
       "      <th>sample_pool</th>\n",
       "      <th>batch</th>\n",
       "      <th>category</th>\n",
       "      <th>gender</th>\n",
       "      <th>age</th>\n",
       "      <th>subtype</th>\n",
       "      <th>treatment</th>\n",
       "      <th>recur</th>\n",
       "      <th>baseline_cancer_type</th>\n",
       "      <th>comment</th>\n",
       "      <th>qc</th>\n",
       "      <th>stage</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>screening_method</th>\n",
       "      <th>ER_status</th>\n",
       "      <th>HER2_status</th>\n",
       "      <th>ER/HER2_method</th>\n",
       "      <th>comorbidity</th>\n",
       "      <th>smoking_status</th>\n",
       "      <th>s3_panelwide_gene</th>\n",
       "      <th>s3_aa</th>\n",
       "      <th>somatic_maxmafpct_panelwide</th>\n",
       "      <th>somatic_source</th>\n",
       "      <th>mcm_gene</th>\n",
       "      <th>mcm_aa</th>\n",
       "      <th>g360_gene</th>\n",
       "      <th>g360_aa</th>\n",
       "      <th>somatic_driver_pct</th>\n",
       "      <th>somatic_driver_gene</th>\n",
       "      <th>hyper_residual_count_pos_ctrl</th>\n",
       "      <th>panel</th>\n",
       "      <th>msi_status</th>\n",
       "      <th>msi_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14125</th>\n",
       "      <td>B079640-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52402.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-12.80378</td>\n",
       "      <td>-9.88626</td>\n",
       "      <td>-12.31540</td>\n",
       "      <td>-5.55190</td>\n",
       "      <td>-5.55190</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100141685</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.150505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/B079640-2.pre-realigned.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0521</td>\n",
       "      <td>other</td>\n",
       "      <td>Female</td>\n",
       "      <td>49.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14147</th>\n",
       "      <td>FB09162087-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>185722.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-7.04119</td>\n",
       "      <td>-5.70565</td>\n",
       "      <td>-9.95082</td>\n",
       "      <td>-3.08714</td>\n",
       "      <td>-7.04119</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ALM_CRC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>106-3061</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.980519</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09162087-2.pre-realigned.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0521</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>45.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14148</th>\n",
       "      <td>FB09162093-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>64462.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-14.99297</td>\n",
       "      <td>-6.19359</td>\n",
       "      <td>-8.86810</td>\n",
       "      <td>-5.36549</td>\n",
       "      <td>-6.19359</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ALM_CRC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>116-3026</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.458196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09162093-2.pre-realigned.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0521</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>51.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14158</th>\n",
       "      <td>FB09630376-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>76634.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-11.49278</td>\n",
       "      <td>-10.28273</td>\n",
       "      <td>-14.57369</td>\n",
       "      <td>-8.91455</td>\n",
       "      <td>-11.49278</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ALM_CRC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>104-3036</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.607516</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09630376-2.pre-realigned.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0521</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14165</th>\n",
       "      <td>FB09630587-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166985.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-3.94195</td>\n",
       "      <td>-3.08528</td>\n",
       "      <td>-8.08071</td>\n",
       "      <td>-2.39462</td>\n",
       "      <td>-3.94195</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ALM_CRC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>132-3012</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.695346</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09630587-2.pre-realigned.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0521</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>78.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample_id runid  ctrl_cor_frac  pos_ctrl_after_correction  \\\n",
       "14125     B079640-2   NaN            1.0                    52402.0   \n",
       "14147  FB09162087-2   NaN            1.0                   185722.0   \n",
       "14148  FB09162093-2   NaN            1.0                    64462.0   \n",
       "14158  FB09630376-2   NaN            1.0                    76634.0   \n",
       "14165  FB09630587-2   NaN            1.0                   166985.0   \n",
       "\n",
       "         pred_type       CRC      LUNG    BREAST  BLADDER  AGGREGATED  CRC_TF  \\\n",
       "14125  cancer_free -12.80378  -9.88626 -12.31540 -5.55190    -5.55190     0.0   \n",
       "14147  cancer_free  -7.04119  -5.70565  -9.95082 -3.08714    -7.04119     0.0   \n",
       "14148  cancer_free -14.99297  -6.19359  -8.86810 -5.36549    -6.19359     0.0   \n",
       "14158  cancer_free -11.49278 -10.28273 -14.57369 -8.91455   -11.49278     0.0   \n",
       "14165  cancer_free  -3.94195  -3.08528  -8.08071 -2.39462    -3.94195     0.0   \n",
       "\n",
       "       LUNG_TF  BREAST_TF  BLADDER_TF  AGGREGATED_TF   cohort request_id  \\\n",
       "14125      0.0        0.0         0.0            0.0   CUB_01        NaN   \n",
       "14147      0.0        0.0         0.0            0.0  ALM_CRC        NaN   \n",
       "14148      0.0        0.0         0.0            0.0  ALM_CRC        NaN   \n",
       "14158      0.0        0.0         0.0            0.0  ALM_CRC        NaN   \n",
       "14165      0.0        0.0         0.0            0.0  ALM_CRC        NaN   \n",
       "\n",
       "      patient_id visit_name  Alt-NoAlt SUCCESS Sample_comment  \\\n",
       "14125  100141685     pre-op        NaN     NaN            NaN   \n",
       "14147   106-3061     pre-op        NaN     NaN            NaN   \n",
       "14148   116-3026     pre-op        NaN     NaN            NaN   \n",
       "14158   104-3036     pre-op        NaN     NaN            NaN   \n",
       "14165   132-3012     pre-op        NaN     NaN            NaN   \n",
       "\n",
       "       max_maf_pct_report RUO_methylation_status  RUO_methylation_tf  \\\n",
       "14125                 NaN                    NaN                 NaN   \n",
       "14147                 NaN                    NaN                 NaN   \n",
       "14148                 NaN                    NaN                 NaN   \n",
       "14158                 NaN                    NaN                 NaN   \n",
       "14165                 NaN                    NaN                 NaN   \n",
       "\n",
       "       TMB_score TMB_category MSI_High   input_ng  Plasma_ml_input  \\\n",
       "14125        NaN          NaN      NaN  22.150505              NaN   \n",
       "14147        NaN          NaN      NaN  44.980519              NaN   \n",
       "14148        NaN          NaN      NaN  17.458196              NaN   \n",
       "14158        NaN          NaN      NaN  22.607516              NaN   \n",
       "14165        NaN          NaN      NaN  44.695346              NaN   \n",
       "\n",
       "      Bloodcoll_date raw_cancer_type  cancer_type  cfDNA_ng_yield subset  \\\n",
       "14125            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "14147            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "14148            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "14158            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "14165            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "\n",
       "      genomic_method fc_name  \\\n",
       "14125         Shield     NaN   \n",
       "14147         Shield     NaN   \n",
       "14148         Shield     NaN   \n",
       "14158         Shield     NaN   \n",
       "14165         Shield     NaN   \n",
       "\n",
       "                                                                                                                                                                           cram_path  \\\n",
       "14125     /ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/B079640-2.pre-realigned.cram   \n",
       "14147  /ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09162087-2.pre-realigned.cram   \n",
       "14148  /ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09162093-2.pre-realigned.cram   \n",
       "14158  /ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09630376-2.pre-realigned.cram   \n",
       "14165  /ghsfa/screening/flowcells/processed/BIP4/devel/220511_A01021_0460_BHHJJCDSX3.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220513_153914/FB09630587-2.pre-realigned.cram   \n",
       "\n",
       "      somatic_max_maf_gene mut_aa  somatic_max_maf_pct is_titration  \\\n",
       "14125                  NaN    NaN                  NaN           no   \n",
       "14147                  NaN    NaN                  NaN           no   \n",
       "14148                  NaN    NaN                  NaN           no   \n",
       "14158                  NaN    NaN                  NaN           no   \n",
       "14165                  NaN    NaN                  NaN           no   \n",
       "\n",
       "      sample_pool     batch category  gender   age subtype  treatment  recur  \\\n",
       "14125         NaN  LTO_0521    other  Female  49.0     NaN        NaN    NaN   \n",
       "14147         NaN  LTO_0521    other     NaN  45.0     NaN        NaN    NaN   \n",
       "14148         NaN  LTO_0521    other     NaN  51.0     NaN        NaN    NaN   \n",
       "14158         NaN  LTO_0521    other     NaN  56.0     NaN        NaN    NaN   \n",
       "14165         NaN  LTO_0521    other     NaN  78.0     NaN        NaN    NaN   \n",
       "\n",
       "       baseline_cancer_type comment       qc    stage ethnicity  \\\n",
       "14125                   NaN     NaN  SUCCESS  grade_6       NaN   \n",
       "14147                   NaN     NaN  SUCCESS      NaN       NaN   \n",
       "14148                   NaN     NaN  SUCCESS      NaN       NaN   \n",
       "14158                   NaN     NaN  SUCCESS      NaN       NaN   \n",
       "14165                   NaN     NaN  SUCCESS      NaN       NaN   \n",
       "\n",
       "      screening_method ER_status HER2_status ER/HER2_method comorbidity  \\\n",
       "14125       colonscopy       NaN         NaN            NaN         NaN   \n",
       "14147              NaN       NaN         NaN            NaN         NaN   \n",
       "14148              NaN       NaN         NaN            NaN         NaN   \n",
       "14158              NaN       NaN         NaN            NaN         NaN   \n",
       "14165              NaN       NaN         NaN            NaN         NaN   \n",
       "\n",
       "      smoking_status s3_panelwide_gene s3_aa  somatic_maxmafpct_panelwide  \\\n",
       "14125            NaN               NaN   NaN                          NaN   \n",
       "14147            NaN               NaN   NaN                          NaN   \n",
       "14148            NaN               NaN   NaN                          NaN   \n",
       "14158            NaN               NaN   NaN                          NaN   \n",
       "14165            NaN               NaN   NaN                          NaN   \n",
       "\n",
       "      somatic_source mcm_gene mcm_aa g360_gene g360_aa  somatic_driver_pct  \\\n",
       "14125            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "14147            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "14148            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "14158            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "14165            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "\n",
       "      somatic_driver_gene  hyper_residual_count_pos_ctrl panel msi_status  \\\n",
       "14125                 NaN                            NaN   MCM        NaN   \n",
       "14147                 NaN                            NaN   MCM        NaN   \n",
       "14148                 NaN                            NaN   MCM        NaN   \n",
       "14158                 NaN                            NaN   MCM        NaN   \n",
       "14165                 NaN                            NaN   MCM        NaN   \n",
       "\n",
       "       msi_score  \n",
       "14125        NaN  \n",
       "14147        NaN  \n",
       "14148        NaN  \n",
       "14158        NaN  \n",
       "14165        NaN  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#to get mcm cancer-free sample (code from Kyle)\n",
    "# meta file of all methyl samples\n",
    "meta_raw = pd.read_csv(\"/ghess/omni_v2/users/schen/methylation_data/all-methyl-eap-dev-training-meta.20230828.all_samples.tsv\", sep=\"\\t\")\n",
    "meta_raw = meta_raw.drop_duplicates(subset=[\"sample_id\"])\n",
    "norm = meta_raw.query('comment.isna() and SUCCESS != \"FAIL\" and cancer_type == \"CANCER_FREE\" and subset==\"mcm\" and cram_path.notna()')\n",
    "norm.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "144940df-55d0-415a-81d8-33431abe209f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4286, 75)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "norm.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a13c7df3-1e50-4cf7-b753-beca4f5a7e83",
   "metadata": {},
   "outputs": [],
   "source": [
    "#edit the 'cram_path' column to generate \"mbd_hyper_file_path\" column for each sample's mbd_hyper_molecules.tsv\n",
    "df_norm = norm.copy()\n",
    "\n",
    "df_norm['mbd_hyper_file_path'] = df_norm['cram_path'].str.replace(r'\\.pre-realigned\\.cram$', '.mbd_hyper_molecules.tsv', regex=True)\n",
    "# df_norm.to_csv('df_norm.csv', index=False)\n",
    "# df_norm.head()\n",
    "\n",
    "# Assuming df_norm is your DataFrame\n",
    "\n",
    "# Calculate the number of rows to split at\n",
    "split_row = len(df_norm) // 2\n",
    "\n",
    "# Split the DataFrame into two halves\n",
    "df_norm_1 = df_norm.iloc[:split_row]\n",
    "df_norm_2 = df_norm.iloc[split_row:]\n",
    "\n",
    "# Save each half to a separate CSV file\n",
    "df_norm_1.to_csv('df_norm_1.csv', index=False)\n",
    "df_norm_2.to_csv('df_norm_2.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bfd5d4a4-2ac1-4d66-8fe9-7a36b97aa75f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#The function to calculate MB score\n",
    "mb_model_coef = pd.read_csv(\"/home/yhe/repo/bip-products/models/Lunar2-0.2/methylation_caller/mafband_v2023_06_30/msre_mr_model.tsv\", sep=\"\\t\")\n",
    "\n",
    "def calculate_MB_score(msre_file, mb_model_coef):\n",
    "    msre_df = pd.read_table(msre_file)\n",
    "    msre_df[\"region_score\"] = np.log10(msre_df[\"hyper_residual_count\"]/msre_df[\"hyper_residual_count_pos_ctrl\"] + 1e-5)\n",
    "    v6_score_df = (\n",
    "        msre_df\n",
    "        .merge(mb_model_coef)\n",
    "        .groupby([\"run_sample_id\", \"runid\"])\n",
    "        .apply(lambda x: (x[\"region_score\"] * x[\"weight\"]).sum() + x[\"bias\"].iloc[0])\n",
    "        .reset_index(name=\"v2_mb_score\")\n",
    "    )\n",
    "    v2_score = v6_score_df.v2_mb_score.values[0]\n",
    "\n",
    "    return v2_score\n",
    "    \n",
    "v2_score = calculate_MB_score(\"/ghess/omni_v2/users/schen/mafband_on_s3/mafband_shield_bip/221208_A01810_0059_AHKW77DSX5_PROD148_AP1.Sirius-1.0.0-RLS/D8_S3P_0988/D8_S3P_0988.msre_caller_mr_features.hdr.tsv.gz\", mb_model_coef)\n",
    "\n",
    "print(v2_score)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a825a5b7-23b6-471f-9578-304730617a70",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_norm = df_norm.head(2500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0245a878-6994-46a0-89ce-967e5a934a39",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# def calculate_methylation_detected(file_path):\n",
    "#     # Check if the file exists\n",
    "#     if not os.path.exists(file_path):\n",
    "#         print(f\"No file found at {file_path}\")\n",
    "#         return np.nan\n",
    "\n",
    "#     try:\n",
    "#         # Read the file into a DataFrame\n",
    "#         df = pd.read_table(file_path)\n",
    "\n",
    "#         # Apply the filtering conditions\n",
    "#         filtered_df = df[(df['chrom'] == 3) &\n",
    "#                          (df['start'] >= 10178462) &\n",
    "#                          (df['start'] <= 10183462) &\n",
    "#                          ((df['end'] - df['start']) < 400) &\n",
    "#                          (df['mean_mapq'] >= 60) &\n",
    "#                          (df['cg_count'] >= 10)]\n",
    "        \n",
    "#         # Calculate the sum of 'num_molecules'\n",
    "#         sum_num_molecules = filtered_df['num_molecules'].sum()\n",
    "\n",
    "#         # Return 1 if sum > 10, else 0\n",
    "#         return 1 if sum_num_molecules > 10 else 0\n",
    "\n",
    "#     except Exception as e:\n",
    "#         print(f\"Error processing file {file_path}: {e}\")\n",
    "#         return np.nan\n",
    "\n",
    "# # Apply the function to each row and assign the result to a new column\n",
    "# df_norm['VHL_promoter_methylation_detected'] = df_norm['mbd_hyper_file_path'].apply(calculate_methylation_detected)\n",
    "# df_norm.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb2b218e-dd75-4acb-94d9-e3a16bcf3b6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_methylation_detected_and_methyl_score(mbd_hyper_file_path, sample_id):\n",
    "    # Check if the mbd_hyper file exists\n",
    "    if not os.path.exists(mbd_hyper_file_path):\n",
    "        print(f\"No file found at {mbd_hyper_file_path}\")\n",
    "        return np.nan, np.nan, np.nan\n",
    "\n",
    "    try:\n",
    "        # Read the mbd_hyper file into a DataFrame\n",
    "        df_mbd_hyper = pd.read_table(mbd_hyper_file_path)\n",
    "\n",
    "        # Apply the primary filtering conditions\n",
    "        filtered_df = df_mbd_hyper[(df_mbd_hyper['chrom'] == 3) &\n",
    "                                   (df_mbd_hyper['start'] >= 10178462) &\n",
    "                                   (df_mbd_hyper['start'] <= 10183462) &\n",
    "                                   ((df_mbd_hyper['end'] - df_mbd_hyper['start']) < 400) &\n",
    "                                   (df_mbd_hyper['mean_mapq'] >= 60) &\n",
    "                                   (df_mbd_hyper['cg_count'] >= 10)]\n",
    "        \n",
    "        # Calculate the sum of 'num_molecules'\n",
    "        sum_num_molecules = filtered_df['num_molecules'].sum()\n",
    "\n",
    "        # Define the path for the secondary file\n",
    "        msre_call_rerun_by_sai = \"/ghess/omni_v2/users/schen/methylation_data/msre_call/\"\n",
    "        secondary_file_name = f\"{sample_id}.msre_all_call.hdr.tsv\"\n",
    "\n",
    "        subfolder = os.popen(f\"ls -d {msre_call_rerun_by_sai}*/{sample_id}\").read().strip()\n",
    "        secondary_file_path = os.path.join(subfolder, secondary_file_name)\n",
    "\n",
    "        # Check if the secondary file exists\n",
    "        if not os.path.exists(secondary_file_path):\n",
    "            print(f\"No secondary file found at {secondary_file_path}\")\n",
    "            return 0, sum_num_molecules, np.nan\n",
    "\n",
    "        # Read the secondary file\n",
    "        df_secondary = pd.read_table(secondary_file_path)\n",
    "        hyper_residual_count_pos_ctrl = df_secondary['hyper_residual_count_pos_ctrl'].iloc[0]\n",
    "\n",
    "        # Calculate methyl_score (just for VHL region)\n",
    "        methyl_score = sum_num_molecules / hyper_residual_count_pos_ctrl if hyper_residual_count_pos_ctrl else np.nan\n",
    "\n",
    "        # Determine the value for 'VHL_promoter_methylation_detected'\n",
    "        methylation_detected = 1 if methyl_score > 8.3e-05 and sum_num_molecules > 10 else 0\n",
    "        return methylation_detected, sum_num_molecules, methyl_score\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"Error processing file {mbd_hyper_file_path}: {e}\")\n",
    "        return np.nan, np.nan, np.nan\n",
    "\n",
    "results = df_norm.apply(lambda row: calculate_methylation_detected_and_methyl_score(row['mbd_hyper_file_path'], row['sample_id']), axis=1)\n",
    "df_norm['VHL_promoter_methylation_detected'], df_norm['sum_VHL_promoter_mol'], df_norm['methyl_score_VHL'] = zip(*results)\n",
    "\n",
    "#save the file as csv\n",
    "df_norm.to_csv('df_norm_vhl_all.csv', index=False)\n",
    "df_norm.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "02cab61e-1a78-4715-8ccc-46844ceb4cad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 70783655\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.system(\"sbatch VHL_promoter_methyl.job\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "291bfb40-f8c0-4bf4-bcfd-26bf8d9f0326",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Writing the function and execution code to a Python script\n",
    "# python_script = \"\"\"\n",
    "# def calculate_methylation_detected_and_methyl_score(mbd_hyper_file_path, sample_id):\n",
    "#     # Check if the mbd_hyper file exists\n",
    "#     if not os.path.exists(mbd_hyper_file_path):\n",
    "#         print(f\"No file found at {mbd_hyper_file_path}\")\n",
    "#         return np.nan, np.nan, np.nan\n",
    "\n",
    "#     try:\n",
    "#         # Read the mbd_hyper file into a DataFrame\n",
    "#         df_mbd_hyper = pd.read_table(mbd_hyper_file_path)\n",
    "\n",
    "#         # Apply the primary filtering conditions\n",
    "#         filtered_df = df_mbd_hyper[(df_mbd_hyper['chrom'] == 3) &\n",
    "#                                    (df_mbd_hyper['start'] >= 10178462) &\n",
    "#                                    (df_mbd_hyper['start'] <= 10183462) &\n",
    "#                                    ((df_mbd_hyper['end'] - df_mbd_hyper['start']) < 400) &\n",
    "#                                    (df_mbd_hyper['mean_mapq'] >= 60) &\n",
    "#                                    (df_mbd_hyper['cg_count'] >= 10)]\n",
    "        \n",
    "#         # Calculate the sum of 'num_molecules'\n",
    "#         sum_num_molecules = filtered_df['num_molecules'].sum()\n",
    "\n",
    "#         # Define the path for the secondary file\n",
    "#         msre_call_rerun_by_sai = \"/ghess/omni_v2/users/schen/methylation_data/msre_call/\"\n",
    "#         secondary_file_name = f\"{sample_id}.msre_all_call.hdr.tsv\"\n",
    "\n",
    "#         subfolder = os.popen(f\"ls -d {msre_call_rerun_by_sai}*/{sample_id}\").read().strip()\n",
    "#         secondary_file_path = os.path.join(subfolder, secondary_file_name)\n",
    "\n",
    "#         # Check if the secondary file exists\n",
    "#         if not os.path.exists(secondary_file_path):\n",
    "#             print(f\"No secondary file found at {secondary_file_path}\")\n",
    "#             return 0, sum_num_molecules, np.nan\n",
    "\n",
    "#         # Read the secondary file\n",
    "#         df_secondary = pd.read_table(secondary_file_path)\n",
    "#         hyper_residual_count_pos_ctrl = df_secondary['hyper_residual_count_pos_ctrl'].iloc[0]\n",
    "\n",
    "#         # Calculate methyl_score\n",
    "#         methyl_score = sum_num_molecules / hyper_residual_count_pos_ctrl if hyper_residual_count_pos_ctrl else np.nan\n",
    "\n",
    "#         # Determine the value for 'VHL_promoter_methylation_detected'\n",
    "#         methylation_detected = 1 if methyl_score > 8.3e-05 and sum_num_molecules > 10 else 0\n",
    "#         return methylation_detected, sum_num_molecules, methyl_score\n",
    "\n",
    "#     except Exception as e:\n",
    "#         print(f\"Error processing file {mbd_hyper_file_path}: {e}\")\n",
    "#         return np.nan, np.nan, np.nan\n",
    "\n",
    "# # Assuming df_norm is already defined or loaded here\n",
    "# df_norm = pd.read_csv('path_to_your_df_norm.csv')\n",
    "\n",
    "# results = df_norm.apply(lambda row: calculate_methylation_detected_and_methyl_score(row['mbd_hyper_file_path'], row['sample_id']), axis=1)\n",
    "# df_norm['VHL_promoter_methylation_detected'], df_norm['sum_VHL_promoter_mol'], df_norm['methyl_score_VHL'] = zip(*results)\n",
    "\n",
    "# df_norm.to_csv('df_norm_vhl_all.csv', index=False)\n",
    "# \"\"\"\n",
    "\n",
    "# with open(\"calculate_methylation_script.py\", \"w\") as file:\n",
    "#     file.write(python_script)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be6f12e5-652d-4438-9dab-7b3d4ed20339",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Writing the Slurm job script\n",
    "# slurm_script = \"\"\"\n",
    "# #!/bin/bash\n",
    "# #SBATCH --job-name=calc_methyl\n",
    "# #SBATCH --output=calc_methyl.log\n",
    "# #SBATCH --nodes=1\n",
    "# #SBATCH --ntasks=1\n",
    "# #SBATCH --time=01:00:00\n",
    "# #SBATCH --mem=10GB\n",
    "\n",
    "# module load python3  # Adjust as needed\n",
    "# python calculate_methylation_script.py\n",
    "# \"\"\"\n",
    "\n",
    "# with open(\"run_calculation.job\", \"w\") as file:\n",
    "#     file.write(slurm_script)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2fe196c2-a0bb-408f-9bc5-824e46aa27d3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>runid</th>\n",
       "      <th>ctrl_cor_frac</th>\n",
       "      <th>pos_ctrl_after_correction</th>\n",
       "      <th>pred_type</th>\n",
       "      <th>CRC</th>\n",
       "      <th>LUNG</th>\n",
       "      <th>BREAST</th>\n",
       "      <th>BLADDER</th>\n",
       "      <th>AGGREGATED</th>\n",
       "      <th>CRC_TF</th>\n",
       "      <th>LUNG_TF</th>\n",
       "      <th>BREAST_TF</th>\n",
       "      <th>BLADDER_TF</th>\n",
       "      <th>AGGREGATED_TF</th>\n",
       "      <th>cohort</th>\n",
       "      <th>request_id</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>Alt-NoAlt</th>\n",
       "      <th>SUCCESS</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>max_maf_pct_report</th>\n",
       "      <th>RUO_methylation_status</th>\n",
       "      <th>RUO_methylation_tf</th>\n",
       "      <th>TMB_score</th>\n",
       "      <th>TMB_category</th>\n",
       "      <th>MSI_High</th>\n",
       "      <th>input_ng</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>raw_cancer_type</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>subset</th>\n",
       "      <th>genomic_method</th>\n",
       "      <th>fc_name</th>\n",
       "      <th>cram_path</th>\n",
       "      <th>somatic_max_maf_gene</th>\n",
       "      <th>mut_aa</th>\n",
       "      <th>somatic_max_maf_pct</th>\n",
       "      <th>is_titration</th>\n",
       "      <th>sample_pool</th>\n",
       "      <th>batch</th>\n",
       "      <th>category</th>\n",
       "      <th>gender</th>\n",
       "      <th>age</th>\n",
       "      <th>subtype</th>\n",
       "      <th>treatment</th>\n",
       "      <th>recur</th>\n",
       "      <th>baseline_cancer_type</th>\n",
       "      <th>comment</th>\n",
       "      <th>qc</th>\n",
       "      <th>stage</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>screening_method</th>\n",
       "      <th>ER_status</th>\n",
       "      <th>HER2_status</th>\n",
       "      <th>ER/HER2_method</th>\n",
       "      <th>comorbidity</th>\n",
       "      <th>smoking_status</th>\n",
       "      <th>s3_panelwide_gene</th>\n",
       "      <th>s3_aa</th>\n",
       "      <th>somatic_maxmafpct_panelwide</th>\n",
       "      <th>somatic_source</th>\n",
       "      <th>mcm_gene</th>\n",
       "      <th>mcm_aa</th>\n",
       "      <th>g360_gene</th>\n",
       "      <th>g360_aa</th>\n",
       "      <th>somatic_driver_pct</th>\n",
       "      <th>somatic_driver_gene</th>\n",
       "      <th>hyper_residual_count_pos_ctrl</th>\n",
       "      <th>panel</th>\n",
       "      <th>msi_status</th>\n",
       "      <th>msi_score</th>\n",
       "      <th>mbd_hyper_file_path</th>\n",
       "      <th>VHL_promoter_methylation_detected</th>\n",
       "      <th>sum_VHL_promoter_mol</th>\n",
       "      <th>methyl_score_VHL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22611</th>\n",
       "      <td>FB09658153-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>98126.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-12.92244</td>\n",
       "      <td>-9.17075</td>\n",
       "      <td>-12.52557</td>\n",
       "      <td>-6.28710</td>\n",
       "      <td>-6.28710</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100225709</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.657933</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.56</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>0.000155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22613</th>\n",
       "      <td>FB09658165-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>148692.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-13.94278</td>\n",
       "      <td>-8.45393</td>\n",
       "      <td>-10.92956</td>\n",
       "      <td>-7.58447</td>\n",
       "      <td>-8.45393</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100224373</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.501710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.25</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22614</th>\n",
       "      <td>FB09658170-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>79510.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-10.62274</td>\n",
       "      <td>-5.30913</td>\n",
       "      <td>-7.66799</td>\n",
       "      <td>-6.45553</td>\n",
       "      <td>-5.30913</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100225676</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.418677</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.39</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>52.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22615</th>\n",
       "      <td>FB09658171-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>152277.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-13.67893</td>\n",
       "      <td>-9.41286</td>\n",
       "      <td>-10.86642</td>\n",
       "      <td>-6.58057</td>\n",
       "      <td>-6.58057</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100242445</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.359027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.15</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22616</th>\n",
       "      <td>FB09658172-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>74112.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-15.36943</td>\n",
       "      <td>-9.93317</td>\n",
       "      <td>-11.72912</td>\n",
       "      <td>-4.97643</td>\n",
       "      <td>-4.97643</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100245566</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.321197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.51</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000151</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample_id runid  ctrl_cor_frac  pos_ctrl_after_correction  \\\n",
       "22611  FB09658153-1   NaN            1.0                    98126.0   \n",
       "22613  FB09658165-1   NaN            1.0                   148692.0   \n",
       "22614  FB09658170-1   NaN            1.0                    79510.0   \n",
       "22615  FB09658171-1   NaN            1.0                   152277.0   \n",
       "22616  FB09658172-1   NaN            1.0                    74112.0   \n",
       "\n",
       "         pred_type       CRC     LUNG    BREAST  BLADDER  AGGREGATED  CRC_TF  \\\n",
       "22611  cancer_free -12.92244 -9.17075 -12.52557 -6.28710    -6.28710     0.0   \n",
       "22613  cancer_free -13.94278 -8.45393 -10.92956 -7.58447    -8.45393     0.0   \n",
       "22614  cancer_free -10.62274 -5.30913  -7.66799 -6.45553    -5.30913     0.0   \n",
       "22615  cancer_free -13.67893 -9.41286 -10.86642 -6.58057    -6.58057     0.0   \n",
       "22616  cancer_free -15.36943 -9.93317 -11.72912 -4.97643    -4.97643     0.0   \n",
       "\n",
       "       LUNG_TF  BREAST_TF  BLADDER_TF  AGGREGATED_TF  cohort request_id  \\\n",
       "22611      0.0        0.0         0.0            0.0  CUB_01        NaN   \n",
       "22613      0.0        0.0         0.0            0.0  CUB_01        NaN   \n",
       "22614      0.0        0.0         0.0            0.0  CUB_01        NaN   \n",
       "22615      0.0        0.0         0.0            0.0  CUB_01        NaN   \n",
       "22616      0.0        0.0         0.0            0.0  CUB_01        NaN   \n",
       "\n",
       "      patient_id visit_name  Alt-NoAlt SUCCESS Sample_comment  \\\n",
       "22611  100225709     pre-op        NaN     NaN            NaN   \n",
       "22613  100224373     pre-op        NaN     NaN            NaN   \n",
       "22614  100225676     pre-op        NaN     NaN            NaN   \n",
       "22615  100242445     pre-op        NaN     NaN            NaN   \n",
       "22616  100245566     pre-op        NaN     NaN            NaN   \n",
       "\n",
       "       max_maf_pct_report RUO_methylation_status  RUO_methylation_tf  \\\n",
       "22611                0.56                    NaN                 NaN   \n",
       "22613                0.25                    NaN                 NaN   \n",
       "22614                0.39                    NaN                 NaN   \n",
       "22615                0.15                    NaN                 NaN   \n",
       "22616                0.51                    NaN                 NaN   \n",
       "\n",
       "       TMB_score TMB_category MSI_High   input_ng  Plasma_ml_input  \\\n",
       "22611        NaN          NaN      NaN  25.657933              NaN   \n",
       "22613        NaN          NaN      NaN  34.501710              NaN   \n",
       "22614        NaN          NaN      NaN  35.418677              NaN   \n",
       "22615        NaN          NaN      NaN  47.359027              NaN   \n",
       "22616        NaN          NaN      NaN  19.321197              NaN   \n",
       "\n",
       "      Bloodcoll_date raw_cancer_type  cancer_type  cfDNA_ng_yield subset  \\\n",
       "22611            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "22613            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "22614            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "22615            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "22616            NaN             NaN  CANCER_FREE             NaN    mcm   \n",
       "\n",
       "      genomic_method fc_name  \\\n",
       "22611         Shield     NaN   \n",
       "22613         Shield     NaN   \n",
       "22614         Shield     NaN   \n",
       "22615         Shield     NaN   \n",
       "22616         Shield     NaN   \n",
       "\n",
       "                                                                                                                                                                                     cram_path  \\\n",
       "22611  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.pre-realigned.cram   \n",
       "22613  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.pre-realigned.cram   \n",
       "22614  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.pre-realigned.cram   \n",
       "22615  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.pre-realigned.cram   \n",
       "22616  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.pre-realigned.cram   \n",
       "\n",
       "      somatic_max_maf_gene        mut_aa  somatic_max_maf_pct is_titration  \\\n",
       "22611                 EGFR  A767_V769dup                 0.56           no   \n",
       "22613                 EGFR  A767_V769dup                 0.25           no   \n",
       "22614                 EGFR  A767_V769dup                 0.39           no   \n",
       "22615                 EGFR  A767_V769dup                 0.15           no   \n",
       "22616                 EGFR  A767_V769dup                 0.51           no   \n",
       "\n",
       "      sample_pool     batch category gender   age subtype  treatment  recur  \\\n",
       "22611         NaN  LTO_0496    other    NaN  56.0     NaN        NaN    NaN   \n",
       "22613         NaN  LTO_0496    other    NaN  46.0     NaN        NaN    NaN   \n",
       "22614         NaN  LTO_0496    other    NaN  52.0     NaN        NaN    NaN   \n",
       "22615         NaN  LTO_0496    other    NaN  65.0     NaN        NaN    NaN   \n",
       "22616         NaN  LTO_0496    other    NaN  46.0     NaN        NaN    NaN   \n",
       "\n",
       "       baseline_cancer_type comment       qc    stage ethnicity  \\\n",
       "22611                   NaN     NaN  SUCCESS  grade_6       NaN   \n",
       "22613                   NaN     NaN  SUCCESS  grade_6       NaN   \n",
       "22614                   NaN     NaN  SUCCESS  grade_6       NaN   \n",
       "22615                   NaN     NaN  SUCCESS  grade_6       NaN   \n",
       "22616                   NaN     NaN  SUCCESS  grade_6       NaN   \n",
       "\n",
       "      screening_method ER_status HER2_status ER/HER2_method comorbidity  \\\n",
       "22611       colonscopy       NaN         NaN            NaN         NaN   \n",
       "22613       colonscopy       NaN         NaN            NaN         NaN   \n",
       "22614       colonscopy       NaN         NaN            NaN         NaN   \n",
       "22615       colonscopy       NaN         NaN            NaN         NaN   \n",
       "22616       colonscopy       NaN         NaN            NaN         NaN   \n",
       "\n",
       "      smoking_status s3_panelwide_gene s3_aa  somatic_maxmafpct_panelwide  \\\n",
       "22611            NaN               NaN   NaN                          NaN   \n",
       "22613            NaN               NaN   NaN                          NaN   \n",
       "22614            NaN               NaN   NaN                          NaN   \n",
       "22615            NaN               NaN   NaN                          NaN   \n",
       "22616            NaN               NaN   NaN                          NaN   \n",
       "\n",
       "      somatic_source mcm_gene mcm_aa g360_gene g360_aa  somatic_driver_pct  \\\n",
       "22611            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "22613            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "22614            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "22615            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "22616            NaN      NaN    NaN       NaN     NaN                 NaN   \n",
       "\n",
       "      somatic_driver_gene  hyper_residual_count_pos_ctrl panel msi_status  \\\n",
       "22611                 NaN                            NaN   MCM        NaN   \n",
       "22613                 NaN                            NaN   MCM        NaN   \n",
       "22614                 NaN                            NaN   MCM        NaN   \n",
       "22615                 NaN                            NaN   MCM        NaN   \n",
       "22616                 NaN                            NaN   MCM        NaN   \n",
       "\n",
       "       msi_score  \\\n",
       "22611        NaN   \n",
       "22613        NaN   \n",
       "22614        NaN   \n",
       "22615        NaN   \n",
       "22616        NaN   \n",
       "\n",
       "                                                                                                                                                                                mbd_hyper_file_path  \\\n",
       "22611  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.mbd_hyper_molecules.tsv   \n",
       "22613  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.mbd_hyper_molecules.tsv   \n",
       "22614  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.mbd_hyper_molecules.tsv   \n",
       "22615  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.mbd_hyper_molecules.tsv   \n",
       "22616  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.mbd_hyper_molecules.tsv   \n",
       "\n",
       "       VHL_promoter_methylation_detected  sum_VHL_promoter_mol  \\\n",
       "22611                                  1                    15   \n",
       "22613                                  0                     3   \n",
       "22614                                  1                    13   \n",
       "22615                                  0                    11   \n",
       "22616                                  1                    11   \n",
       "\n",
       "       methyl_score_VHL  \n",
       "22611          0.000155  \n",
       "22613          0.000020  \n",
       "22614          0.000167  \n",
       "22615          0.000074  \n",
       "22616          0.000151  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# def calculate_methylation_detected_and_methyl_score(mbd_hyper_file_path, sample_id):\n",
    "#     # Check if the mbd_hyper file exists\n",
    "#     if not os.path.exists(mbd_hyper_file_path):\n",
    "#         print(f\"No file found at {mbd_hyper_file_path}\")\n",
    "#         return np.nan, np.nan, np.nan\n",
    "\n",
    "#     try:\n",
    "#         # Read the mbd_hyper file into a DataFrame\n",
    "#         df_mbd_hyper = pd.read_table(mbd_hyper_file_path)\n",
    "\n",
    "#         # Apply the primary filtering conditions\n",
    "#         filtered_df = df_mbd_hyper[(df_mbd_hyper['chrom'] == 3) &\n",
    "#                                    (df_mbd_hyper['start'] >= 10178462) &\n",
    "#                                    (df_mbd_hyper['start'] <= 10183462) &\n",
    "#                                    ((df_mbd_hyper['end'] - df_mbd_hyper['start']) < 400) &\n",
    "#                                    (df_mbd_hyper['mean_mapq'] >= 60) &\n",
    "#                                    (df_mbd_hyper['cg_count'] >= 10)]\n",
    "        \n",
    "#         # Calculate the sum of 'num_molecules'\n",
    "#         sum_num_molecules = filtered_df['num_molecules'].sum()\n",
    "\n",
    "#         # Define the path for the secondary file\n",
    "#         msre_call_rerun_by_sai = \"/ghess/omni_v2/users/schen/methylation_data/msre_call/\"\n",
    "#         secondary_file_name = f\"{sample_id}.msre_all_call.hdr.tsv\"\n",
    "\n",
    "#         subfolder = os.popen(f\"ls -d {msre_call_rerun_by_sai}*/{sample_id}\").read().strip()\n",
    "#         secondary_file_path = os.path.join(subfolder, secondary_file_name)\n",
    "\n",
    "#         # Check if the secondary file exists\n",
    "#         if not os.path.exists(secondary_file_path):\n",
    "#             print(f\"No secondary file found at {secondary_file_path}\")\n",
    "#             return 0, sum_num_molecules, np.nan\n",
    "\n",
    "#         # Read the secondary file\n",
    "#         df_secondary = pd.read_table(secondary_file_path)\n",
    "#         hyper_residual_count_pos_ctrl = df_secondary['hyper_residual_count_pos_ctrl'].iloc[0]\n",
    "\n",
    "#         # Calculate methyl_score\n",
    "#         methyl_score = sum_num_molecules / hyper_residual_count_pos_ctrl if hyper_residual_count_pos_ctrl else np.nan\n",
    "\n",
    "#         # Determine the value for 'VHL_promoter_methylation_detected'\n",
    "#         methylation_detected = 1 if methyl_score > 8.3e-05 and sum_num_molecules > 10 else 0\n",
    "#         return methylation_detected, sum_num_molecules, methyl_score\n",
    "\n",
    "#     except Exception as e:\n",
    "#         print(f\"Error processing file {mbd_hyper_file_path}: {e}\")\n",
    "#         return np.nan, np.nan, np.nan\n",
    "\n",
    "# # Function to process a chunk of the DataFrame\n",
    "# def process_df_chunk(df_chunk):\n",
    "#     results = df_chunk.apply(lambda row: calculate_methylation_detected_and_methyl_score(row['mbd_hyper_file_path'], row['sample_id']), axis=1)\n",
    "#     df_chunk['VHL_promoter_methylation_detected'], df_chunk['sum_VHL_promoter_mol'], df_chunk['methyl_score_VHL'] = zip(*results)\n",
    "#     return df_chunk\n",
    "\n",
    "# # Function to process the DataFrame in batches\n",
    "# def process_in_batches(df, batch_size):\n",
    "#     result_frames = []\n",
    "#     for start in range(0, len(df), batch_size):\n",
    "#         end = min(start + batch_size, len(df))\n",
    "#         df_batch = df[start:end]\n",
    "#         processed_batch = process_df_chunk(df_batch)\n",
    "#         result_frames.append(processed_batch)\n",
    "#     return pd.concat(result_frames)\n",
    "\n",
    "# # Define batch size\n",
    "# batch_size = 500 \n",
    "\n",
    "# # Process df_norm in batches\n",
    "# df_norm_processed = process_in_batches(df_norm, batch_size)\n",
    "\n",
    "# # Save the final DataFrame\n",
    "# df_norm_processed.to_csv('df_norm_vhl_tail_500.csv', index=False)\n",
    "\n",
    "# # Display the head of the processed DataFrame\n",
    "# df_norm_processed.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "291cfc9f-6265-4387-9c5b-94cb59a8c933",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_3169/1687523304.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m \u001b[0;31m# Process df_norm in batches and save each batch\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 72\u001b[0;31m \u001b[0mprocess_in_batches_and_save\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_norm\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch_size\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/tmp/ipykernel_3169/1687523304.py\u001b[0m in \u001b[0;36mprocess_in_batches_and_save\u001b[0;34m(df, batch_size)\u001b[0m\n\u001b[1;32m     59\u001b[0m         \u001b[0mend\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstart\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mbatch_size\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m         \u001b[0mdf_batch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0mprocessed_batch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprocess_df_chunk\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_batch\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m         \u001b[0;31m# Save each batch to a separate CSV file\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/tmp/ipykernel_3169/1687523304.py\u001b[0m in \u001b[0;36mprocess_df_chunk\u001b[0;34m(df_chunk)\u001b[0m\n\u001b[1;32m     50\u001b[0m \u001b[0;31m# Function to process a chunk of the DataFrame\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     51\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mprocess_df_chunk\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_chunk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 52\u001b[0;31m     \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcalculate_methylation_detected_and_methyl_score\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'mbd_hyper_file_path'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sample_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     53\u001b[0m     \u001b[0mdf_chunk\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'VHL_promoter_methylation_detected'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sum_VHL_promoter_mol'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'methyl_score_VHL'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mzip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mresults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, func, axis, raw, result_type, args, **kwargs)\u001b[0m\n\u001b[1;32m   9556\u001b[0m             \u001b[0mkwargs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   9557\u001b[0m         )\n\u001b[0;32m-> 9558\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__finalize__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"apply\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   9559\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   9560\u001b[0m     def applymap(\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    739\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_raw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    740\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 741\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    742\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    743\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0magg\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    866\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    867\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 868\u001b[0;31m         \u001b[0mresults\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mres_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_series_generator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    869\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    870\u001b[0m         \u001b[0;31m# wrap results\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_series_generator\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    882\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mv\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries_gen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    883\u001b[0m                 \u001b[0;31m# ignore SettingWithCopy here in case the user mutates\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 884\u001b[0;31m                 \u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mv\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    885\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mABCSeries\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    886\u001b[0m                     \u001b[0;31m# If we have a view on v, we need to make a copy because\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/tmp/ipykernel_3169/1687523304.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(row)\u001b[0m\n\u001b[1;32m     50\u001b[0m \u001b[0;31m# Function to process a chunk of the DataFrame\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     51\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mprocess_df_chunk\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_chunk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 52\u001b[0;31m     \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcalculate_methylation_detected_and_methyl_score\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'mbd_hyper_file_path'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sample_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     53\u001b[0m     \u001b[0mdf_chunk\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'VHL_promoter_methylation_detected'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sum_VHL_promoter_mol'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'methyl_score_VHL'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mzip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mresults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mdf_chunk\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/tmp/ipykernel_3169/1687523304.py\u001b[0m in \u001b[0;36mcalculate_methylation_detected_and_methyl_score\u001b[0;34m(mbd_hyper_file_path, sample_id)\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m         \u001b[0;31m# Read the mbd_hyper file into a DataFrame\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m         \u001b[0mdf_mbd_hyper\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_table\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmbd_hyper_file_path\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m         \u001b[0;31m# Apply the primary filtering conditions\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    209\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m                     \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnew_arg_name\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_arg_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    212\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mF\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    315\u001b[0m                     \u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfind_stack_level\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minspect\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcurrentframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    316\u001b[0m                 )\n\u001b[0;32m--> 317\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    318\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    319\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread_table\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[1;32m   1287\u001b[0m     \u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwds_defaults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1289\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1290\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1291\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    609\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    610\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mparser\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 611\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    612\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    613\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, nrows)\u001b[0m\n\u001b[1;32m   1787\u001b[0m                 \u001b[0mnew_rows\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1788\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1789\u001b[0;31m             \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcol_dict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1790\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1791\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_currow\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mnew_rows\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    661\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    662\u001b[0m             \u001b[0;31m# GH#38939 de facto copy defaults to False only in non-dict cases\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 663\u001b[0;31m             \u001b[0mmgr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdict_to_mgr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtyp\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmanager\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    664\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mma\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMaskedArray\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    665\u001b[0m             \u001b[0;32mimport\u001b[0m \u001b[0mnumpy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mma\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmrecords\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mmrecords\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36mdict_to_mgr\u001b[0;34m(data, index, columns, dtype, typ, copy)\u001b[0m\n\u001b[1;32m    492\u001b[0m             \u001b[0marrays\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mhasattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"dtype\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0marrays\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    493\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 494\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0marrays_to_mgr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtyp\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtyp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconsolidate\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    495\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    496\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36marrays_to_mgr\u001b[0;34m(arrays, columns, index, dtype, verify_integrity, typ, consolidate)\u001b[0m\n\u001b[1;32m    153\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    154\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mtyp\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"block\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 155\u001b[0;31m         return create_block_manager_from_column_arrays(\n\u001b[0m\u001b[1;32m    156\u001b[0m             \u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconsolidate\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mconsolidate\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    157\u001b[0m         )\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mcreate_block_manager_from_column_arrays\u001b[0;34m(arrays, axes, consolidate)\u001b[0m\n\u001b[1;32m   2164\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2165\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2166\u001b[0;31m         \u001b[0mblocks\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_form_blocks\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconsolidate\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2167\u001b[0m         \u001b[0mmgr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mBlockManager\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mblocks\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverify_integrity\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2168\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mValueError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36m_form_blocks\u001b[0;34m(arrays, consolidate)\u001b[0m\n\u001b[1;32m   2238\u001b[0m                 \u001b[0mdtype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2239\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2240\u001b[0;31m             \u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mplacement\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_stack_arrays\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtup_block\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2241\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_dtlike\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2242\u001b[0m                 \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_wrapped_if_datetimelike\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/lunartk_v1.1.0/lib/python3.8/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36m_stack_arrays\u001b[0;34m(tuples, dtype)\u001b[0m\n\u001b[1;32m   2279\u001b[0m     \u001b[0mstacked\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mempty\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2280\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marr\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marrays\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2281\u001b[0;31m         \u001b[0mstacked\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2282\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2283\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mstacked\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mplacement\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# # Function to calculate methylation detected and methyl score\n",
    "# def calculate_methylation_detected_and_methyl_score(mbd_hyper_file_path, sample_id):\n",
    "#     # Check if the mbd_hyper file exists\n",
    "#     if not os.path.exists(mbd_hyper_file_path):\n",
    "#         print(f\"No file found at {mbd_hyper_file_path}\")\n",
    "#         return np.nan, np.nan, np.nan\n",
    "\n",
    "#     try:\n",
    "#         # Read the mbd_hyper file into a DataFrame\n",
    "#         df_mbd_hyper = pd.read_table(mbd_hyper_file_path)\n",
    "\n",
    "#         # Apply the primary filtering conditions\n",
    "#         filtered_df = df_mbd_hyper[(df_mbd_hyper['chrom'] == 3) &\n",
    "#                                    (df_mbd_hyper['start'] >= 10178462) &\n",
    "#                                    (df_mbd_hyper['start'] <= 10183462) &\n",
    "#                                    ((df_mbd_hyper['end'] - df_mbd_hyper['start']) < 400) &\n",
    "#                                    (df_mbd_hyper['mean_mapq'] >= 60) &\n",
    "#                                    (df_mbd_hyper['cg_count'] >= 10)]\n",
    "        \n",
    "#         # Calculate the sum of 'num_molecules'\n",
    "#         sum_num_molecules = filtered_df['num_molecules'].sum()\n",
    "\n",
    "#         # Define the path for the secondary file\n",
    "#         msre_call_rerun_by_sai = \"/ghess/omni_v2/users/schen/methylation_data/msre_call/\"\n",
    "#         secondary_file_name = f\"{sample_id}.msre_all_call.hdr.tsv\"\n",
    "\n",
    "#         subfolder = os.popen(f\"ls -d {msre_call_rerun_by_sai}*/{sample_id}\").read().strip()\n",
    "#         secondary_file_path = os.path.join(subfolder, secondary_file_name)\n",
    "\n",
    "#         # Check if the secondary file exists\n",
    "#         if not os.path.exists(secondary_file_path):\n",
    "#             print(f\"No secondary file found at {secondary_file_path}\")\n",
    "#             return 0, sum_num_molecules, np.nan\n",
    "\n",
    "#         # Read the secondary file\n",
    "#         df_secondary = pd.read_table(secondary_file_path)\n",
    "#         hyper_residual_count_pos_ctrl = df_secondary['hyper_residual_count_pos_ctrl'].iloc[0]\n",
    "\n",
    "#         # Calculate methyl_score\n",
    "#         methyl_score = sum_num_molecules / hyper_residual_count_pos_ctrl if hyper_residual_count_pos_ctrl else np.nan\n",
    "\n",
    "#         # Determine the value for 'VHL_promoter_methylation_detected'\n",
    "#         methylation_detected = 1 if methyl_score > 8.3e-05 and sum_num_molecules > 10 else 0\n",
    "#         return methylation_detected, sum_num_molecules, methyl_score\n",
    "\n",
    "#     except Exception as e:\n",
    "#         print(f\"Error processing file {mbd_hyper_file_path}: {e}\")\n",
    "#         return np.nan, np.nan, np.nan\n",
    "\n",
    "# # Function to process a chunk of the DataFrame\n",
    "# def process_df_chunk(df_chunk):\n",
    "#     results = df_chunk.apply(lambda row: calculate_methylation_detected_and_methyl_score(row['mbd_hyper_file_path'], row['sample_id']), axis=1)\n",
    "#     df_chunk['VHL_promoter_methylation_detected'], df_chunk['sum_VHL_promoter_mol'], df_chunk['methyl_score_VHL'] = zip(*results)\n",
    "#     return df_chunk\n",
    "\n",
    "# # Function to process the DataFrame in batches and save each batch\n",
    "# def process_in_batches_and_save(df, batch_size):\n",
    "#     for i, start in enumerate(range(0, len(df), batch_size)):\n",
    "#         end = min(start + batch_size, len(df))\n",
    "#         df_batch = df[start:end]\n",
    "#         processed_batch = process_df_chunk(df_batch)\n",
    "        \n",
    "#         # Save each batch to a separate CSV file\n",
    "#         batch_file_name = f'df_norm_vhl_batch_{i}.csv'\n",
    "#         processed_batch.to_csv(batch_file_name, index=False)\n",
    "#         print(f\"Batch {i} saved to {batch_file_name}\")\n",
    "\n",
    "# # Define batch size\n",
    "# batch_size = 500 \n",
    "\n",
    "# # Process df_norm in batches and save each batch\n",
    "# process_in_batches_and_save(df_norm, batch_size)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f091c6f7-9322-4248-b83c-81af0052288b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Percentage of 1s in 'VHL_promoter_methylation_detected': 19.80%\n",
      "Percentage of mol > 10 in 'sum_VHL_promoter_mol': 33.00%\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('df_norm_vhl_tail_500.csv', header=0)\n",
    "\n",
    "# Calculate the percentage of 1s in the 'VHL_promoter_methylation_detected' column\n",
    "percentage_of_ones = (df['VHL_promoter_methylation_detected'].sum() / len(df)) * 100\n",
    "percentage_greater_than_10 = (df['sum_VHL_promoter_mol'] > 10).sum() / len(df) * 100\n",
    "\n",
    "print(f\"Percentage of 1s in 'VHL_promoter_methylation_detected': {percentage_of_ones:.2f}%\")\n",
    "print(f\"Percentage of mol > 10 in 'sum_VHL_promoter_mol': {percentage_greater_than_10:.2f}%\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "954c2569-d8c3-4ba5-a3a3-f6bdfb1d1866",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import epicaller\n",
    "# import epicaller.methylation.methyl_caller\n",
    "\n",
    "# def run_mb_crc_caller_wrapper_slurm(sample_meta_df, rundir):\n",
    "#     region_file='/ghds/groups/labdesk/bshih/clinical_genomics/bims_testing/bi-variant-monitoring/rundeck_monitoring_scripts/CRC_methyl_caller_files/v05_classification_space.tsv'\n",
    "#     control_region_file='/ghds/groups/labdesk/bshih/clinical_genomics/bims_testing/bi-variant-monitoring/rundeck_monitoring_scripts/CRC_methyl_caller_files/v05_positive_control_regions.tsv'\n",
    "#     lr_model_file='/ghds/groups/labdesk/bshih/clinical_genomics/bims_testing/bi-variant-monitoring/rundeck_monitoring_scripts/CRC_methyl_caller_files/msre_lr_model.tsv'\n",
    "#     mr_model_file='/ghds/groups/labdesk/bshih/clinical_genomics/bims_testing/bi-variant-monitoring/rundeck_monitoring_scripts/CRC_methyl_caller_files/msre_mr_model.tsv'\n",
    "    \n",
    "#     for _, row in sample_meta_df.iterrows():\n",
    "#         run_sample_id = row[\"run_sample_id\"]\n",
    "#         fc_dir = row[\"fc_dir\"]\n",
    "#         flowcell_id = row[\"runid\"]\n",
    "#         mbd_hyper_molecules = f\"{fc_dir}/{run_sample_id}/{run_sample_id}.mbd_hyper_molecules.tsv\"\n",
    "#         # mbd_residual_molecules = f\"{fc_dir}/{run_sample_id}/{run_sample_id}.mbd_residual_molecules.tsv\"\n",
    "        \n",
    "#         # Check if the molecule file exists\n",
    "#         if not os.path.exists(mbd_hyper_molecules):\n",
    "#             continue\n",
    "        \n",
    "#         # Check if the job folder exists\n",
    "#         if not(os.path.exists(f\"{rundir}/MB_CRC_jobs\")):\n",
    "#             os.mkdir(f\"{rundir}/MB_CRC_jobs\")\n",
    "\n",
    "#         job_file = f\"{rundir}/MB_CRC_jobs/MB_call_{run_sample_id}.job\" \n",
    "#         shebang_content = (f\"#!/bin/bash\\n\"\n",
    "#                            f\"#SBATCH --job-name=MB_call_{run_sample_id}\\n\"\n",
    "#                            f\"#SBATCH --nodes=1\\n\"\n",
    "#                            f\"#SBATCH --ntasks=1\\n\"\n",
    "#                            f\"#SBATCH --cpus-per-task=1\\n\"\n",
    "#                            f\"#SBATCH --mem=30gb\\n\"\n",
    "#                            f\"#SBATCH --partition=dev\\n\"\n",
    "#                            f\"#SBATCH --time=00:10:00\\n\"\n",
    "#                            f\"#SBATCH --output={rundir}/MB_CRC_jobs/MB_call_{run_sample_id}.log\\n\\n\")\n",
    "        \n",
    "#         with open(job_file, \"w\") as job_output:\n",
    "#             job_output.write(shebang_content)\n",
    "#             job_output.write(\"python3 -m epicaller.methylation.methyl_caller \" + \n",
    "#                    f\"--hyper-molecule-file {mbd_hyper_molecules} \" + \n",
    "#                    f\"--region-file {region_file} \" +\n",
    "#                    f\"--control-region-file {control_region_file} \" + \n",
    "#                    f\"--lr-model-file {lr_model_file} \" + \n",
    "#                    f\"--mr-model-file {mr_model_file} \" +\n",
    "#                    \"--min-frag-len 120 --max-frag-len 240 --min-cg-count 12 --max-cg-count 30 --cutsite-count-cutoff 1 --min-mapq 60 \" + \n",
    "#                    f\"--sample-id {run_sample_id} --flowcell-id {runid} --outdir {rundir}/\") \n",
    "#             # print(job_output)    \n",
    "#         os.system(f\"sbatch {job_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "acede229-8440-4ed6-8478-7872946c60d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>runid</th>\n",
       "      <th>ctrl_cor_frac</th>\n",
       "      <th>pos_ctrl_after_correction</th>\n",
       "      <th>pred_type</th>\n",
       "      <th>CRC</th>\n",
       "      <th>LUNG</th>\n",
       "      <th>BREAST</th>\n",
       "      <th>BLADDER</th>\n",
       "      <th>AGGREGATED</th>\n",
       "      <th>CRC_TF</th>\n",
       "      <th>LUNG_TF</th>\n",
       "      <th>BREAST_TF</th>\n",
       "      <th>BLADDER_TF</th>\n",
       "      <th>AGGREGATED_TF</th>\n",
       "      <th>cohort</th>\n",
       "      <th>request_id</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>Alt-NoAlt</th>\n",
       "      <th>SUCCESS</th>\n",
       "      <th>Sample_comment</th>\n",
       "      <th>max_maf_pct_report</th>\n",
       "      <th>RUO_methylation_status</th>\n",
       "      <th>RUO_methylation_tf</th>\n",
       "      <th>TMB_score</th>\n",
       "      <th>TMB_category</th>\n",
       "      <th>MSI_High</th>\n",
       "      <th>input_ng</th>\n",
       "      <th>Plasma_ml_input</th>\n",
       "      <th>Bloodcoll_date</th>\n",
       "      <th>raw_cancer_type</th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>cfDNA_ng_yield</th>\n",
       "      <th>subset</th>\n",
       "      <th>genomic_method</th>\n",
       "      <th>fc_name</th>\n",
       "      <th>cram_path</th>\n",
       "      <th>somatic_max_maf_gene</th>\n",
       "      <th>mut_aa</th>\n",
       "      <th>somatic_max_maf_pct</th>\n",
       "      <th>is_titration</th>\n",
       "      <th>sample_pool</th>\n",
       "      <th>batch</th>\n",
       "      <th>category</th>\n",
       "      <th>gender</th>\n",
       "      <th>age</th>\n",
       "      <th>subtype</th>\n",
       "      <th>treatment</th>\n",
       "      <th>recur</th>\n",
       "      <th>baseline_cancer_type</th>\n",
       "      <th>comment</th>\n",
       "      <th>qc</th>\n",
       "      <th>stage</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>screening_method</th>\n",
       "      <th>ER_status</th>\n",
       "      <th>HER2_status</th>\n",
       "      <th>ER/HER2_method</th>\n",
       "      <th>comorbidity</th>\n",
       "      <th>smoking_status</th>\n",
       "      <th>s3_panelwide_gene</th>\n",
       "      <th>s3_aa</th>\n",
       "      <th>somatic_maxmafpct_panelwide</th>\n",
       "      <th>somatic_source</th>\n",
       "      <th>mcm_gene</th>\n",
       "      <th>mcm_aa</th>\n",
       "      <th>g360_gene</th>\n",
       "      <th>g360_aa</th>\n",
       "      <th>somatic_driver_pct</th>\n",
       "      <th>somatic_driver_gene</th>\n",
       "      <th>hyper_residual_count_pos_ctrl</th>\n",
       "      <th>panel</th>\n",
       "      <th>msi_status</th>\n",
       "      <th>msi_score</th>\n",
       "      <th>mbd_hyper_file_path</th>\n",
       "      <th>VHL_promoter_methylation_detected</th>\n",
       "      <th>sum_VHL_promoter_mol</th>\n",
       "      <th>methyl_score_VHL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FB09658153-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>98126.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-12.92244</td>\n",
       "      <td>-9.17075</td>\n",
       "      <td>-12.52557</td>\n",
       "      <td>-6.28710</td>\n",
       "      <td>-6.28710</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100225709</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.657933</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.56</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>56.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>0.000155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FB09658165-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>148692.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-13.94278</td>\n",
       "      <td>-8.45393</td>\n",
       "      <td>-10.92956</td>\n",
       "      <td>-7.58447</td>\n",
       "      <td>-8.45393</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100224373</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.501710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.25</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FB09658170-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>79510.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-10.62274</td>\n",
       "      <td>-5.30913</td>\n",
       "      <td>-7.66799</td>\n",
       "      <td>-6.45553</td>\n",
       "      <td>-5.30913</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100225676</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.418677</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.39</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>52.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FB09658171-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>152277.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-13.67893</td>\n",
       "      <td>-9.41286</td>\n",
       "      <td>-10.86642</td>\n",
       "      <td>-6.58057</td>\n",
       "      <td>-6.58057</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100242445</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.359027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.15</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FB09658172-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>74112.0</td>\n",
       "      <td>cancer_free</td>\n",
       "      <td>-15.36943</td>\n",
       "      <td>-9.93317</td>\n",
       "      <td>-11.72912</td>\n",
       "      <td>-4.97643</td>\n",
       "      <td>-4.97643</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CUB_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100245566</td>\n",
       "      <td>pre-op</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.321197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CANCER_FREE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mcm</td>\n",
       "      <td>Shield</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.pre-realigned.cram</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>0.51</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LTO_0496</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>46.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>grade_6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colonscopy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.mbd_hyper_molecules.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000151</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample_id  runid  ctrl_cor_frac  pos_ctrl_after_correction    pred_type  \\\n",
       "0  FB09658153-1    NaN            1.0                    98126.0  cancer_free   \n",
       "1  FB09658165-1    NaN            1.0                   148692.0  cancer_free   \n",
       "2  FB09658170-1    NaN            1.0                    79510.0  cancer_free   \n",
       "3  FB09658171-1    NaN            1.0                   152277.0  cancer_free   \n",
       "4  FB09658172-1    NaN            1.0                    74112.0  cancer_free   \n",
       "\n",
       "        CRC     LUNG    BREAST  BLADDER  AGGREGATED  CRC_TF  LUNG_TF  \\\n",
       "0 -12.92244 -9.17075 -12.52557 -6.28710    -6.28710     0.0      0.0   \n",
       "1 -13.94278 -8.45393 -10.92956 -7.58447    -8.45393     0.0      0.0   \n",
       "2 -10.62274 -5.30913  -7.66799 -6.45553    -5.30913     0.0      0.0   \n",
       "3 -13.67893 -9.41286 -10.86642 -6.58057    -6.58057     0.0      0.0   \n",
       "4 -15.36943 -9.93317 -11.72912 -4.97643    -4.97643     0.0      0.0   \n",
       "\n",
       "   BREAST_TF  BLADDER_TF  AGGREGATED_TF  cohort  request_id patient_id  \\\n",
       "0        0.0         0.0            0.0  CUB_01         NaN  100225709   \n",
       "1        0.0         0.0            0.0  CUB_01         NaN  100224373   \n",
       "2        0.0         0.0            0.0  CUB_01         NaN  100225676   \n",
       "3        0.0         0.0            0.0  CUB_01         NaN  100242445   \n",
       "4        0.0         0.0            0.0  CUB_01         NaN  100245566   \n",
       "\n",
       "  visit_name  Alt-NoAlt  SUCCESS  Sample_comment  max_maf_pct_report  \\\n",
       "0     pre-op        NaN      NaN             NaN                0.56   \n",
       "1     pre-op        NaN      NaN             NaN                0.25   \n",
       "2     pre-op        NaN      NaN             NaN                0.39   \n",
       "3     pre-op        NaN      NaN             NaN                0.15   \n",
       "4     pre-op        NaN      NaN             NaN                0.51   \n",
       "\n",
       "   RUO_methylation_status  RUO_methylation_tf  TMB_score  TMB_category  \\\n",
       "0                     NaN                 NaN        NaN           NaN   \n",
       "1                     NaN                 NaN        NaN           NaN   \n",
       "2                     NaN                 NaN        NaN           NaN   \n",
       "3                     NaN                 NaN        NaN           NaN   \n",
       "4                     NaN                 NaN        NaN           NaN   \n",
       "\n",
       "   MSI_High   input_ng  Plasma_ml_input  Bloodcoll_date  raw_cancer_type  \\\n",
       "0       NaN  25.657933              NaN             NaN              NaN   \n",
       "1       NaN  34.501710              NaN             NaN              NaN   \n",
       "2       NaN  35.418677              NaN             NaN              NaN   \n",
       "3       NaN  47.359027              NaN             NaN              NaN   \n",
       "4       NaN  19.321197              NaN             NaN              NaN   \n",
       "\n",
       "   cancer_type  cfDNA_ng_yield subset genomic_method  fc_name  \\\n",
       "0  CANCER_FREE             NaN    mcm         Shield      NaN   \n",
       "1  CANCER_FREE             NaN    mcm         Shield      NaN   \n",
       "2  CANCER_FREE             NaN    mcm         Shield      NaN   \n",
       "3  CANCER_FREE             NaN    mcm         Shield      NaN   \n",
       "4  CANCER_FREE             NaN    mcm         Shield      NaN   \n",
       "\n",
       "                                                                                                                                                                                 cram_path  \\\n",
       "0  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.pre-realigned.cram   \n",
       "1  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.pre-realigned.cram   \n",
       "2  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.pre-realigned.cram   \n",
       "3  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.pre-realigned.cram   \n",
       "4  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.pre-realigned.cram   \n",
       "\n",
       "  somatic_max_maf_gene        mut_aa  somatic_max_maf_pct is_titration  \\\n",
       "0                 EGFR  A767_V769dup                 0.56           no   \n",
       "1                 EGFR  A767_V769dup                 0.25           no   \n",
       "2                 EGFR  A767_V769dup                 0.39           no   \n",
       "3                 EGFR  A767_V769dup                 0.15           no   \n",
       "4                 EGFR  A767_V769dup                 0.51           no   \n",
       "\n",
       "   sample_pool     batch category gender   age  subtype  treatment  recur  \\\n",
       "0          NaN  LTO_0496    other    NaN  56.0      NaN        NaN    NaN   \n",
       "1          NaN  LTO_0496    other    NaN  46.0      NaN        NaN    NaN   \n",
       "2          NaN  LTO_0496    other    NaN  52.0      NaN        NaN    NaN   \n",
       "3          NaN  LTO_0496    other    NaN  65.0      NaN        NaN    NaN   \n",
       "4          NaN  LTO_0496    other    NaN  46.0      NaN        NaN    NaN   \n",
       "\n",
       "   baseline_cancer_type  comment       qc    stage ethnicity screening_method  \\\n",
       "0                   NaN      NaN  SUCCESS  grade_6       NaN       colonscopy   \n",
       "1                   NaN      NaN  SUCCESS  grade_6       NaN       colonscopy   \n",
       "2                   NaN      NaN  SUCCESS  grade_6       NaN       colonscopy   \n",
       "3                   NaN      NaN  SUCCESS  grade_6       NaN       colonscopy   \n",
       "4                   NaN      NaN  SUCCESS  grade_6       NaN       colonscopy   \n",
       "\n",
       "   ER_status  HER2_status  ER/HER2_method comorbidity smoking_status  \\\n",
       "0        NaN          NaN             NaN         NaN            NaN   \n",
       "1        NaN          NaN             NaN         NaN            NaN   \n",
       "2        NaN          NaN             NaN         NaN            NaN   \n",
       "3        NaN          NaN             NaN         NaN            NaN   \n",
       "4        NaN          NaN             NaN         NaN            NaN   \n",
       "\n",
       "   s3_panelwide_gene  s3_aa  somatic_maxmafpct_panelwide somatic_source  \\\n",
       "0                NaN    NaN                          NaN            NaN   \n",
       "1                NaN    NaN                          NaN            NaN   \n",
       "2                NaN    NaN                          NaN            NaN   \n",
       "3                NaN    NaN                          NaN            NaN   \n",
       "4                NaN    NaN                          NaN            NaN   \n",
       "\n",
       "  mcm_gene mcm_aa  g360_gene  g360_aa  somatic_driver_pct somatic_driver_gene  \\\n",
       "0      NaN    NaN        NaN      NaN                 NaN                 NaN   \n",
       "1      NaN    NaN        NaN      NaN                 NaN                 NaN   \n",
       "2      NaN    NaN        NaN      NaN                 NaN                 NaN   \n",
       "3      NaN    NaN        NaN      NaN                 NaN                 NaN   \n",
       "4      NaN    NaN        NaN      NaN                 NaN                 NaN   \n",
       "\n",
       "   hyper_residual_count_pos_ctrl panel  msi_status  msi_score  \\\n",
       "0                            NaN   MCM         NaN        NaN   \n",
       "1                            NaN   MCM         NaN        NaN   \n",
       "2                            NaN   MCM         NaN        NaN   \n",
       "3                            NaN   MCM         NaN        NaN   \n",
       "4                            NaN   MCM         NaN        NaN   \n",
       "\n",
       "                                                                                                                                                                            mbd_hyper_file_path  \\\n",
       "0  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658153-1.mbd_hyper_molecules.tsv   \n",
       "1  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658165-1.mbd_hyper_molecules.tsv   \n",
       "2  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658170-1.mbd_hyper_molecules.tsv   \n",
       "3  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658171-1.mbd_hyper_molecules.tsv   \n",
       "4  /screening/archive/groups/lunar/flowcells/processed/devel/220415_A01194_0214_AH7MVCDMXY.devel_methylome_v2_8.MethylomeT123v2_set6barcodes.220418_130205/FB09658172-1.mbd_hyper_molecules.tsv   \n",
       "\n",
       "   VHL_promoter_methylation_detected  sum_VHL_promoter_mol  methyl_score_VHL  \n",
       "0                                  1                    15          0.000155  \n",
       "1                                  0                     3          0.000020  \n",
       "2                                  1                    13          0.000167  \n",
       "3                                  0                    11          0.000074  \n",
       "4                                  1                    11          0.000151  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# def submit_slurm_job(notebook_path, output_log_dir, job_name=\"notebook_job\", time_limit=\"00:10:00\", mem=\"30gb\"):\n",
    "#     # Create job script content\n",
    "#     job_script = f\"\"\"#!/bin/bash\n",
    "# #SBATCH --job-name={job_name}\n",
    "# #SBATCH --output={output_log_dir}/{job_name}.log\n",
    "# #SBATCH --nodes=1\n",
    "# #SBATCH --ntasks=1\n",
    "# #SBATCH --cpus-per-task=1\n",
    "# #SBATCH --mem={mem}\n",
    "# #SBATCH --time={time_limit}\n",
    "# #SBATCH --partition=dev\n",
    "\n",
    "# module load jupyter  # Load the necessary module, adjust if needed\n",
    "# jupyter nbconvert --to notebook --execute {notebook_path} --output {notebook_path}_executed.ipynb\n",
    "# \"\"\"\n",
    "\n",
    "#     # Create a temporary job script file\n",
    "#     job_script_file = f\"{job_name}.job\"\n",
    "#     with open(job_script_file, \"w\") as file:\n",
    "#         file.write(job_script)\n",
    "\n",
    "#     # Submit the job script to Slurm\n",
    "#     os.system(f\"sbatch {job_script_file}\")\n",
    "\n",
    "# # Path to your Jupyter Notebook\n",
    "# notebook_path = \"VHL_promoter_methylation_12122023.ipynb\"\n",
    "\n",
    "# # Directory where you want to save the Slurm log files\n",
    "# output_log_dir = \"/home/hazhang/projects/projects/VHL_promoter_methylation_12122023\"\n",
    "\n",
    "# # Submit the job\n",
    "# submit_slurm_job(notebook_path, output_log_dir)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9f69cd1-1b26-41bc-868d-1c3f3383ccbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# #used this as WTF to find all the S3 fc_dir as the original WTF script missed some folder\n",
    "# import pandas as pd\n",
    "# import glob\n",
    "\n",
    "# # Read the CSV file\n",
    "# ghdb = pd.read_csv(f\"{work_dir}/ghdb.csv\", header=0)\n",
    "\n",
    "# # Define the path to the 'flowcentral' directory\n",
    "# fc_all_folder = \"/ghds/ivd/flowcentral\"\n",
    "\n",
    "# # Add a new column 'fc_dir' to the DataFrame with default empty values\n",
    "# ghdb['fc_dir'] = ''\n",
    "\n",
    "# # Iterate over each row of the DataFrame\n",
    "# for index, row in ghdb.iterrows():\n",
    "#     try:\n",
    "#         runid = row[\"runid\"]\n",
    "#         pattern = f\"{fc_all_folder}/*{runid}.*\"\n",
    "        \n",
    "#         # Use the glob function to find folders that match the pattern\n",
    "#         folders = glob.glob(pattern)\n",
    "        \n",
    "#         # Check if any matching folders were found\n",
    "#         if not folders:\n",
    "#             print(f\"No suitable folder found for {runid}\")\n",
    "#             ghdb.loc[index, \"fc_dir\"] = ''\n",
    "#         else:\n",
    "#             folder = folders[0]\n",
    "#             ghdb.loc[index, \"fc_dir\"] = folder\n",
    "\n",
    "#     except Exception as e:\n",
    "#         print(f\"An error occurred while processing {runid}: {str(e)}\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:lunartk_v1.1.0] *",
   "language": "python",
   "name": "conda-env-lunartk_v1.1.0-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
